<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Ashenafee Mandefro" />

<meta name="date" content="2024-03-12" />

<title>Assignment 2</title>

<script src="A2_Notebook_files/header-attrs-2.25/header-attrs.js"></script>
<script src="A2_Notebook_files/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="A2_Notebook_files/bootstrap-3.3.5/css/journal.min.css" rel="stylesheet" />
<script src="A2_Notebook_files/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="A2_Notebook_files/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="A2_Notebook_files/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="A2_Notebook_files/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="A2_Notebook_files/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="A2_Notebook_files/tocify-1.9.1/jquery.tocify.js"></script>
<script src="A2_Notebook_files/navigation-1.1/tabsets.js"></script>
<script src="A2_Notebook_files/navigation-1.1/codefolding.js"></script>
<link href="A2_Notebook_files/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="A2_Notebook_files/pagedtable-1.1/js/pagedtable.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #204a87; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #8f5902; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #204a87; font-weight: bold; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #ce5c00; font-weight: bold; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

.sourceCode .row {
  width: 100%;
}
.sourceCode {
  overflow-x: auto;
}
.code-folding-btn {
  margin-right: -30px;
}
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>



<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>



<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Assignment 2</h1>
<h3 class="subtitle">Analyzing Gene Expression Trends in Cannabidiol
(CBD) Treated Cells on SARS-CoV-2 Infection</h3>
<h4 class="author">Ashenafee Mandefro</h4>
<h4 class="date">03/12/2024</h4>

</div>


<div id="introduction" class="section level1">
<h1>Introduction</h1>
<div id="background" class="section level2 tabset tabset-fade">
<h2 class="tabset tabset-fade">Background</h2>
<div id="phytocannabinoids" class="section level3">
<h3>Phytocannabinoids</h3>
<p>A phytocannabinoid is a cannabinoid that occurs naturally in plants,
such as Cannabis <em>sativa</em>. The word “phyto” means “plant”, and
the prefix “phyto-” is used in biochemistry to refer to compounds
derived from plants. These compounds are known to occur in several plant
species besides cannabis. The distinction with the prefix “phyto-” must
be made because the human body also produces cannabinoids, referred to
as <em>endo</em>cannabinoids due to their endogenous nature.</p>
</div>
<div id="cannabidiol-cbd" class="section level3">
<h3>Cannabidiol (CBD)</h3>
<p>Cannabidiol (CBD) is one of the numerous phytocannabinoids found in
the Cannabis (Cannabis sativa) plant. However, it is not responsible for
the “high” commonly associated with use of cannabis - that
responsibility falls on delta-9 tetrohydrocannabinol (THC). CBD is a
non-psychoactive compound, and has been shown to have a wide range of
potential therapeutic properties, including anti-inflammatory,
antioxidant, and neuroprotective effects<span
class="citation"><sup>1,2</sup></span>. It has showed promise as a
possible treatment for a variety of conditions, including epilepsy,
anxiety, and chronic pain<span class="citation"><sup>3</sup></span>.</p>
</div>
<div id="sars-cov-2" class="section level3">
<h3>SARS-CoV-2</h3>
<p>A highly contagious virus, SARS-CoV-2 is the driving force behind the
COVID-19 pandemic. It is a novel coronavirus that was first identified
in December 2019 in Wuhan, China, and is primarily spread through
respiratory droplets. Those who have been infected with the virus can
experience a wide range of symptoms, from mild to severe, with the worst
outcome being death. Due to its extreme impact on public health, global
economy, and daily life, significant amounts of research went into
treatment/prevention methods for the virus from 2020 to 2022.</p>
</div>
</div>
<div id="primary-research" class="section level2">
<h2>Primary Research</h2>
<div id="quick-summary-of-the-paper" class="section level3">
<h3>Quick Summary of the Paper</h3>
<p><a
href="https://www.science.org/doi/10.1126/sciadv.abi6110">Cannabidiol
Inhibits SARS-CoV-2 Replication through Induction of the Host ER Stress
and Innate Immune Responses</a> discusses the effects of CBD treatment
in SARS-CoV-2 infected cells and mice, through its action of preventing
replication of the virus. The researchers found that these effects are
apparent in lung epithelial cells which is a critical finding, given
that SARS-CoV-2 manifests itself with many respiratory issues<span
class="citation"><sup>4</sup></span>.</p>
</div>
<div id="data-overview" class="section level3">
<h3>Data Overview</h3>
<p>The RNASeq data from this paper is available on the <a
href="https://www.ncbi.nlm.nih.gov/geo/">Gene Expression Omnibus
(GEO)</a> under the accession number <a
href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE168797">GSE168797</a>.
More information about the data is summarized in the table below:</p>
<table>
<colgroup>
<col width="53%" />
<col width="46%" />
</colgroup>
<thead>
<tr class="header">
<th>Attribute</th>
<th>Value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Title</td>
<td>Cannabidiol Inhibits SARS-CoV-2 Replication through Induction of the
Host ER Stress and Innate Immune Responses</td>
</tr>
<tr class="even">
<td>Organism</td>
<td>Homo sapiens</td>
</tr>
<tr class="odd">
<td>Paper</td>
<td>Nguyen, L. C., Yang, D., Nicolaescu, V., Best, T. J., Gula, H.,
Saxena, D., Gabbard, J. D., Chen, S. N., Ohtsuki, T., Friesen, J. B.,
Drayman, N., Mohamed, A., Dann, C., Silva, D., Robinson-Mailman, L.,
Valdespino, A., Stock, L., Suárez, E., Jones, K. A., Azizi, S. A., …
Rosner, M. R. (2022). Cannabidiol inhibits SARS-CoV-2 replication
through induction of the host ER stress and innate immune responses.
Science advances, 8(8), eabi6110. <a
href="https://doi.org/10.1126/sciadv.abi6110"
class="uri">https://doi.org/10.1126/sciadv.abi6110</a></td>
</tr>
<tr class="even">
<td>Samples</td>
<td>24</td>
</tr>
<tr class="odd">
<td>Status</td>
<td>Public on Jan 20, 2022</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="data-preprocessing" class="section level1">
<h1>Data Preprocessing</h1>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="co"># Load the RData file</span></span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;filtered_data_matrix.RData&quot;</span>)</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;samples_type.RData&quot;</span>)</span></code></pre></div>
<div id="data-cleaning" class="section level2">
<h2>Data Cleaning</h2>
<p>The initial phase of data preprocessing consisted of multiple layers
of data cleaning. The raw dataset from GEO had 57 832 entries but after
cleaning, that was reduced to 20438 entries. These &gt;20 000 entries
were then used in subsequent steps for data normalization and
visualization.</p>
</div>
<div id="data-normalization" class="section level2 tabset tabset-fade">
<h2 class="tabset tabset-fade">Data Normalization</h2>
<p>The cleaned data was visualized using both a boxplot and a density
plot to get a quick overview of the distribution of the data and see if
things were indeed normalized successfully.</p>
<div id="boxplot" class="section level3">
<h3>Boxplot</h3>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="co"># Define the log2 TPM</span></span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a>lg2_tpm <span class="ot">&lt;-</span> <span class="fu">log2</span>(filtered_data_matrix <span class="sc">+</span> <span class="dv">1</span>)</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a><span class="co"># Set the boxplot colors</span></span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a>box_colors <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">&quot;red&quot;</span>, <span class="dv">6</span>), <span class="fu">rep</span>(<span class="st">&quot;blue&quot;</span>, <span class="dv">6</span>))</span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a></span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a><span class="co"># Plot the normalized data</span></span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a><span class="fu">boxplot</span>(</span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a>    lg2_tpm,</span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a>    <span class="at">xaxt =</span> <span class="st">&quot;n&quot;</span>,</span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a>    <span class="at">las =</span> <span class="dv">2</span>,</span>
<span id="cb2-12"><a href="#cb2-12" tabindex="-1"></a>    <span class="at">main =</span> <span class="st">&quot;Cannabidiol as COVID Treatment RNASeq Boxplot (Normalized)&quot;</span>,</span>
<span id="cb2-13"><a href="#cb2-13" tabindex="-1"></a>    <span class="at">ylab =</span> <span class="st">&quot;Log2 TPM&quot;</span>,</span>
<span id="cb2-14"><a href="#cb2-14" tabindex="-1"></a>    <span class="at">xlab =</span> <span class="st">&quot;Treatment&quot;</span>,</span>
<span id="cb2-15"><a href="#cb2-15" tabindex="-1"></a>    <span class="at">cex =</span> <span class="fl">0.5</span>,</span>
<span id="cb2-16"><a href="#cb2-16" tabindex="-1"></a>    <span class="at">cex.lab =</span> <span class="fl">0.8</span>,</span>
<span id="cb2-17"><a href="#cb2-17" tabindex="-1"></a>    <span class="at">col =</span> box_colors</span>
<span id="cb2-18"><a href="#cb2-18" tabindex="-1"></a>)</span>
<span id="cb2-19"><a href="#cb2-19" tabindex="-1"></a></span>
<span id="cb2-20"><a href="#cb2-20" tabindex="-1"></a><span class="co"># Draw the median line</span></span>
<span id="cb2-21"><a href="#cb2-21" tabindex="-1"></a><span class="fu">abline</span>(</span>
<span id="cb2-22"><a href="#cb2-22" tabindex="-1"></a>    <span class="at">h =</span> <span class="fu">median</span>(<span class="fu">apply</span>(lg2_tpm, <span class="dv">2</span>, median)),</span>
<span id="cb2-23"><a href="#cb2-23" tabindex="-1"></a>    <span class="at">col =</span> <span class="st">&quot;green&quot;</span>,</span>
<span id="cb2-24"><a href="#cb2-24" tabindex="-1"></a>    <span class="at">lty =</span> <span class="dv">2</span>,</span>
<span id="cb2-25"><a href="#cb2-25" tabindex="-1"></a>    <span class="at">lwd =</span> <span class="dv">2</span></span>
<span id="cb2-26"><a href="#cb2-26" tabindex="-1"></a>)</span>
<span id="cb2-27"><a href="#cb2-27" tabindex="-1"></a></span>
<span id="cb2-28"><a href="#cb2-28" tabindex="-1"></a><span class="co"># Add x-axis labels manually with a 45 degree rotation</span></span>
<span id="cb2-29"><a href="#cb2-29" tabindex="-1"></a><span class="fu">text</span>(</span>
<span id="cb2-30"><a href="#cb2-30" tabindex="-1"></a>    <span class="at">x =</span> <span class="fu">seq_along</span>(lg2_tpm),</span>
<span id="cb2-31"><a href="#cb2-31" tabindex="-1"></a>    <span class="at">y =</span> <span class="fu">par</span>(<span class="st">&quot;usr&quot;</span>)[<span class="dv">3</span>] <span class="sc">-</span> <span class="fl">0.2</span>,</span>
<span id="cb2-32"><a href="#cb2-32" tabindex="-1"></a>    <span class="at">adj =</span> <span class="dv">1</span>,</span>
<span id="cb2-33"><a href="#cb2-33" tabindex="-1"></a>    <span class="at">srt =</span> <span class="dv">45</span>,</span>
<span id="cb2-34"><a href="#cb2-34" tabindex="-1"></a>    <span class="at">labels =</span> <span class="fu">colnames</span>(lg2_tpm),</span>
<span id="cb2-35"><a href="#cb2-35" tabindex="-1"></a>    <span class="at">xpd =</span> <span class="cn">TRUE</span>,</span>
<span id="cb2-36"><a href="#cb2-36" tabindex="-1"></a>    <span class="at">cex =</span> <span class="fl">0.8</span></span>
<span id="cb2-37"><a href="#cb2-37" tabindex="-1"></a>)</span>
<span id="cb2-38"><a href="#cb2-38" tabindex="-1"></a></span>
<span id="cb2-39"><a href="#cb2-39" tabindex="-1"></a><span class="co"># Add a legend (red - SAR-CoV-2, blue - Mock)</span></span>
<span id="cb2-40"><a href="#cb2-40" tabindex="-1"></a><span class="fu">legend</span>(</span>
<span id="cb2-41"><a href="#cb2-41" tabindex="-1"></a>    <span class="st">&quot;topright&quot;</span>,</span>
<span id="cb2-42"><a href="#cb2-42" tabindex="-1"></a>    <span class="at">legend =</span> <span class="fu">c</span>(<span class="st">&quot;SARS-CoV-2&quot;</span>, <span class="st">&quot;Mock&quot;</span>),</span>
<span id="cb2-43"><a href="#cb2-43" tabindex="-1"></a>    <span class="at">fill =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>)</span>
<span id="cb2-44"><a href="#cb2-44" tabindex="-1"></a>)</span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="A2_Notebook_files/figure-html/boxplot-1.png" alt="Boxplot of the normalized data" width="100%" />
<p class="caption">
Boxplot of the normalized data
</p>
</div>
</div>
<div id="density-plot" class="section level3">
<h3>Density Plot</h3>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="co"># Density plot</span></span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>counts_density <span class="ot">&lt;-</span> <span class="fu">apply</span>(lg2_tpm, <span class="dv">2</span>, density)</span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a>xlim <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a>ylim <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">seq_along</span>(counts_density)) {</span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a>    xlim <span class="ot">&lt;-</span> <span class="fu">range</span>(<span class="fu">c</span>(xlim, counts_density[[i]]<span class="sc">$</span>x))</span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a>    ylim <span class="ot">&lt;-</span> <span class="fu">range</span>(<span class="fu">c</span>(ylim, counts_density[[i]]<span class="sc">$</span>y))</span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a>}</span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a></span>
<span id="cb3-10"><a href="#cb3-10" tabindex="-1"></a>cols <span class="ot">&lt;-</span> <span class="fu">rainbow</span>(<span class="fu">length</span>(counts_density))</span>
<span id="cb3-11"><a href="#cb3-11" tabindex="-1"></a>ltys <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="dv">1</span>, <span class="fu">length</span>(counts_density))</span>
<span id="cb3-12"><a href="#cb3-12" tabindex="-1"></a><span class="fu">plot</span>(</span>
<span id="cb3-13"><a href="#cb3-13" tabindex="-1"></a>    counts_density[[<span class="dv">1</span>]],</span>
<span id="cb3-14"><a href="#cb3-14" tabindex="-1"></a>    <span class="at">xlim =</span> xlim,</span>
<span id="cb3-15"><a href="#cb3-15" tabindex="-1"></a>    <span class="at">ylim =</span> ylim,</span>
<span id="cb3-16"><a href="#cb3-16" tabindex="-1"></a>    <span class="at">type =</span> <span class="st">&quot;n&quot;</span>,</span>
<span id="cb3-17"><a href="#cb3-17" tabindex="-1"></a>    <span class="at">ylab =</span> <span class="st">&quot;Smoothing density of log2-CPM&quot;</span>,</span>
<span id="cb3-18"><a href="#cb3-18" tabindex="-1"></a>    <span class="at">main =</span> <span class="st">&quot;&quot;</span>,</span>
<span id="cb3-19"><a href="#cb3-19" tabindex="-1"></a>    <span class="at">cex.lab =</span> <span class="fl">0.85</span></span>
<span id="cb3-20"><a href="#cb3-20" tabindex="-1"></a>)</span>
<span id="cb3-21"><a href="#cb3-21" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">seq_along</span>(counts_density)) {</span>
<span id="cb3-22"><a href="#cb3-22" tabindex="-1"></a>    <span class="fu">lines</span>(counts_density[[i]], <span class="at">col =</span> cols[i], <span class="at">lty =</span> ltys[i])</span>
<span id="cb3-23"><a href="#cb3-23" tabindex="-1"></a>}</span>
<span id="cb3-24"><a href="#cb3-24" tabindex="-1"></a></span>
<span id="cb3-25"><a href="#cb3-25" tabindex="-1"></a><span class="fu">legend</span>(</span>
<span id="cb3-26"><a href="#cb3-26" tabindex="-1"></a>    <span class="st">&quot;topright&quot;</span>,</span>
<span id="cb3-27"><a href="#cb3-27" tabindex="-1"></a>    <span class="at">legend =</span> <span class="fu">colnames</span>(lg2_tpm),</span>
<span id="cb3-28"><a href="#cb3-28" tabindex="-1"></a>    <span class="at">col =</span> cols,</span>
<span id="cb3-29"><a href="#cb3-29" tabindex="-1"></a>    <span class="at">lty =</span> ltys,</span>
<span id="cb3-30"><a href="#cb3-30" tabindex="-1"></a>    <span class="at">cex =</span> <span class="fl">0.8</span></span>
<span id="cb3-31"><a href="#cb3-31" tabindex="-1"></a>)</span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="A2_Notebook_files/figure-html/densityplot-1.png" alt="Density plot of the normalized data" width="100%" />
<p class="caption">
Density plot of the normalized data
</p>
</div>
</div>
</div>
</div>
<div id="differential-expression-analysis" class="section level1">
<h1>Differential Expression Analysis</h1>
<div id="mds-plot" class="section level2 tabset tabset-fade">
<h2 class="tabset tabset-fade">MDS Plot</h2>
<div id="by-treatment" class="section level3">
<h3>By Treatment</h3>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mar =</span> <span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">4</span>, <span class="dv">4</span>, <span class="dv">15</span>) <span class="sc">+</span> <span class="fl">0.5</span>)</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a><span class="co"># Create the edgeR container for the count data</span></span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a>d <span class="ot">&lt;-</span> edgeR<span class="sc">::</span><span class="fu">DGEList</span>(</span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a>    <span class="at">counts =</span> <span class="fu">as.matrix</span>(filtered_data_matrix),</span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a>    <span class="at">group =</span> samples_type<span class="sc">$</span>treatment</span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a>)</span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a></span>
<span id="cb4-9"><a href="#cb4-9" tabindex="-1"></a><span class="co"># Create a vector of symbols</span></span>
<span id="cb4-10"><a href="#cb4-10" tabindex="-1"></a>symbols <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">factor</span>(<span class="fu">sub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">..*&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="fu">rownames</span>(d<span class="sc">$</span>samples))))</span>
<span id="cb4-11"><a href="#cb4-11" tabindex="-1"></a></span>
<span id="cb4-12"><a href="#cb4-12" tabindex="-1"></a><span class="co"># Perform MDS analysis for treatment</span></span>
<span id="cb4-13"><a href="#cb4-13" tabindex="-1"></a>edgeR<span class="sc">::</span><span class="fu">plotMDS.DGEList</span>(</span>
<span id="cb4-14"><a href="#cb4-14" tabindex="-1"></a>    d,</span>
<span id="cb4-15"><a href="#cb4-15" tabindex="-1"></a>    <span class="at">labels =</span> <span class="cn">NULL</span>,</span>
<span id="cb4-16"><a href="#cb4-16" tabindex="-1"></a>    <span class="at">pch =</span> symbols,</span>
<span id="cb4-17"><a href="#cb4-17" tabindex="-1"></a>    <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>)[<span class="fu">factor</span>(samples_type<span class="sc">$</span>treatment)],</span>
<span id="cb4-18"><a href="#cb4-18" tabindex="-1"></a>)</span>
<span id="cb4-19"><a href="#cb4-19" tabindex="-1"></a></span>
<span id="cb4-20"><a href="#cb4-20" tabindex="-1"></a><span class="co"># Add a title and legend for treatment</span></span>
<span id="cb4-21"><a href="#cb4-21" tabindex="-1"></a><span class="fu">title</span>(<span class="at">main =</span> <span class="st">&quot;MDS Plot by Treatment&quot;</span>)</span>
<span id="cb4-22"><a href="#cb4-22" tabindex="-1"></a></span>
<span id="cb4-23"><a href="#cb4-23" tabindex="-1"></a><span class="co"># Add a legend beside the plot</span></span>
<span id="cb4-24"><a href="#cb4-24" tabindex="-1"></a><span class="fu">legend</span>(</span>
<span id="cb4-25"><a href="#cb4-25" tabindex="-1"></a>    <span class="st">&quot;topright&quot;</span>,</span>
<span id="cb4-26"><a href="#cb4-26" tabindex="-1"></a>    <span class="at">inset=</span><span class="fu">c</span>(<span class="sc">-</span><span class="fl">0.75</span>,<span class="fl">0.4</span>),</span>
<span id="cb4-27"><a href="#cb4-27" tabindex="-1"></a>    <span class="at">legend =</span> <span class="fu">levels</span>(<span class="fu">factor</span>(<span class="fu">sub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">..*&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="fu">rownames</span>(d<span class="sc">$</span>samples)))),</span>
<span id="cb4-28"><a href="#cb4-28" tabindex="-1"></a>    <span class="at">pch =</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(<span class="fu">levels</span>(<span class="fu">factor</span>(<span class="fu">sub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">..*&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="fu">rownames</span>(d<span class="sc">$</span>samples))))),</span>
<span id="cb4-29"><a href="#cb4-29" tabindex="-1"></a>    <span class="at">fill =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>),</span>
<span id="cb4-30"><a href="#cb4-30" tabindex="-1"></a>    <span class="at">xpd =</span> <span class="cn">NA</span></span>
<span id="cb4-31"><a href="#cb4-31" tabindex="-1"></a>)</span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="A2_Notebook_files/figure-html/mdsplot_treatment-1.png" alt="MDS Plot by Treatment of the Filtered Data" width="100%" />
<p class="caption">
MDS Plot by Treatment of the Filtered Data
</p>
</div>
</div>
<div id="by-infection" class="section level3">
<h3>By Infection</h3>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mar =</span> <span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">4</span>, <span class="dv">4</span>, <span class="dv">15</span>) <span class="sc">+</span> <span class="fl">0.5</span>)</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a></span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a><span class="co"># Create the edgeR container for the count data</span></span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a>d <span class="ot">&lt;-</span> edgeR<span class="sc">::</span><span class="fu">DGEList</span>(</span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a>    <span class="at">counts =</span> <span class="fu">as.matrix</span>(filtered_data_matrix),</span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a>    <span class="at">group =</span> samples_type<span class="sc">$</span>infection</span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a>)</span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a></span>
<span id="cb5-9"><a href="#cb5-9" tabindex="-1"></a><span class="co"># Create a vector of symbols</span></span>
<span id="cb5-10"><a href="#cb5-10" tabindex="-1"></a>symbols <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">factor</span>(<span class="fu">sub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">..*&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="fu">rownames</span>(d<span class="sc">$</span>samples))))</span>
<span id="cb5-11"><a href="#cb5-11" tabindex="-1"></a></span>
<span id="cb5-12"><a href="#cb5-12" tabindex="-1"></a><span class="co"># Perform MDS analysis for treatment</span></span>
<span id="cb5-13"><a href="#cb5-13" tabindex="-1"></a>edgeR<span class="sc">::</span><span class="fu">plotMDS.DGEList</span>(</span>
<span id="cb5-14"><a href="#cb5-14" tabindex="-1"></a>    d,</span>
<span id="cb5-15"><a href="#cb5-15" tabindex="-1"></a>    <span class="at">labels =</span> <span class="cn">NULL</span>,</span>
<span id="cb5-16"><a href="#cb5-16" tabindex="-1"></a>    <span class="at">pch =</span> symbols,</span>
<span id="cb5-17"><a href="#cb5-17" tabindex="-1"></a>    <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>)[<span class="fu">factor</span>(samples_type<span class="sc">$</span>infection)],</span>
<span id="cb5-18"><a href="#cb5-18" tabindex="-1"></a>)</span>
<span id="cb5-19"><a href="#cb5-19" tabindex="-1"></a></span>
<span id="cb5-20"><a href="#cb5-20" tabindex="-1"></a><span class="co"># Add a title and legend for infection</span></span>
<span id="cb5-21"><a href="#cb5-21" tabindex="-1"></a><span class="fu">title</span>(<span class="at">main =</span> <span class="st">&quot;MDS Plot by Infection&quot;</span>)</span>
<span id="cb5-22"><a href="#cb5-22" tabindex="-1"></a></span>
<span id="cb5-23"><a href="#cb5-23" tabindex="-1"></a><span class="co"># Add a legend beside the plot</span></span>
<span id="cb5-24"><a href="#cb5-24" tabindex="-1"></a><span class="fu">legend</span>(</span>
<span id="cb5-25"><a href="#cb5-25" tabindex="-1"></a>    <span class="st">&quot;topright&quot;</span>,</span>
<span id="cb5-26"><a href="#cb5-26" tabindex="-1"></a>    <span class="at">inset=</span><span class="fu">c</span>(<span class="sc">-</span><span class="fl">0.75</span>,<span class="fl">0.4</span>),</span>
<span id="cb5-27"><a href="#cb5-27" tabindex="-1"></a>    <span class="at">legend =</span> <span class="fu">levels</span>(<span class="fu">factor</span>(<span class="fu">sub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">..*&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="fu">rownames</span>(d<span class="sc">$</span>samples)))),</span>
<span id="cb5-28"><a href="#cb5-28" tabindex="-1"></a>    <span class="at">pch =</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(<span class="fu">levels</span>(<span class="fu">factor</span>(<span class="fu">sub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">..*&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="fu">rownames</span>(d<span class="sc">$</span>samples))))),</span>
<span id="cb5-29"><a href="#cb5-29" tabindex="-1"></a>    <span class="at">fill =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>),</span>
<span id="cb5-30"><a href="#cb5-30" tabindex="-1"></a>    <span class="at">xpd =</span> <span class="cn">NA</span></span>
<span id="cb5-31"><a href="#cb5-31" tabindex="-1"></a>)</span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="A2_Notebook_files/figure-html/mdsplot_infection-1.png" alt="MDS Plot by Infection of the Filtered Data" width="100%" />
<p class="caption">
MDS Plot by Infection of the Filtered Data
</p>
</div>
</div>
</div>
<div id="dispersion-and-mean-variance-plot"
class="section level2 tabset tabset-fade">
<h2 class="tabset tabset-fade">Dispersion and Mean-Variance Plot</h2>
<div id="dispersion-plot" class="section level3">
<h3>Dispersion Plot</h3>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="co"># Create the design matrix</span></span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a>design <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span> samples_type<span class="sc">$</span>infection <span class="sc">+</span> samples_type<span class="sc">$</span>treatment)</span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a>d <span class="ot">&lt;-</span> edgeR<span class="sc">::</span><span class="fu">estimateDisp</span>(d, design)</span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a></span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a><span class="co"># Plot the dispersion</span></span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a>edgeR<span class="sc">::</span><span class="fu">plotBCV</span>(d, <span class="at">col.tagwise =</span> <span class="st">&quot;black&quot;</span>, <span class="at">col.common =</span> <span class="st">&quot;red&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="A2_Notebook_files/figure-html/dispersion-1.png" alt="Dispersion Plot" width="100%" />
<p class="caption">
Dispersion Plot
</p>
</div>
</div>
<div id="mean-variance-plot" class="section level3">
<h3>Mean-Variance Plot</h3>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="co"># Plot the mean-variance relationship</span></span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a>edgeR<span class="sc">::</span><span class="fu">plotMeanVar</span>(</span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a>    d,</span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a>    <span class="at">show.raw.vars =</span> <span class="cn">TRUE</span>,</span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a>    <span class="at">show.tagwise.vars =</span> <span class="cn">TRUE</span>,</span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a>    <span class="at">NBline =</span> <span class="cn">TRUE</span>,</span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a>    <span class="at">show.ave.raw.vars =</span> <span class="cn">TRUE</span>,</span>
<span id="cb7-8"><a href="#cb7-8" tabindex="-1"></a>    <span class="at">show.binned.common.disp.vars =</span> <span class="cn">TRUE</span></span>
<span id="cb7-9"><a href="#cb7-9" tabindex="-1"></a>)</span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="A2_Notebook_files/figure-html/meanvariance-1.png" alt="Mean-Variance Plot" width="100%" />
<p class="caption">
Mean-Variance Plot
</p>
</div>
</div>
</div>
<div id="examining-different-conditions"
class="section level2 tabset tabset-fade">
<h2 class="tabset tabset-fade">Examining Different Conditions</h2>
<div id="cbd-healthy-vs-dmso-healthy" class="section level3">
<h3>CBD Healthy vs DMSO Healthy</h3>
<div id="gene-counts" class="section level4">
<h4>Gene Counts</h4>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="co"># Subset the data</span></span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a>healthy_data_matrix <span class="ot">&lt;-</span> filtered_data_matrix[, <span class="fu">grepl</span>(<span class="st">&quot;Veh_mock|CBD_mock&quot;</span>, <span class="fu">colnames</span>(filtered_data_matrix))]</span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a>group <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">&quot;CBD&quot;</span>, <span class="dv">3</span>), <span class="fu">rep</span>(<span class="st">&quot;DMSO&quot;</span>, <span class="dv">3</span>)))</span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a></span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a><span class="co"># Create the DGEList object and the design matrix</span></span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a>d <span class="ot">&lt;-</span> edgeR<span class="sc">::</span><span class="fu">DGEList</span>(</span>
<span id="cb8-7"><a href="#cb8-7" tabindex="-1"></a>    <span class="at">counts =</span> <span class="fu">as.matrix</span>(healthy_data_matrix),</span>
<span id="cb8-8"><a href="#cb8-8" tabindex="-1"></a>    <span class="at">group =</span> group</span>
<span id="cb8-9"><a href="#cb8-9" tabindex="-1"></a>)</span>
<span id="cb8-10"><a href="#cb8-10" tabindex="-1"></a>design <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span>group)</span>
<span id="cb8-11"><a href="#cb8-11" tabindex="-1"></a></span>
<span id="cb8-12"><a href="#cb8-12" tabindex="-1"></a><span class="co"># Estimate the dispersion and fit the model</span></span>
<span id="cb8-13"><a href="#cb8-13" tabindex="-1"></a>d <span class="ot">&lt;-</span> edgeR<span class="sc">::</span><span class="fu">estimateDisp</span>(d, design)</span>
<span id="cb8-14"><a href="#cb8-14" tabindex="-1"></a>fit <span class="ot">&lt;-</span> edgeR<span class="sc">::</span><span class="fu">glmQLFit</span>(d, design)</span>
<span id="cb8-15"><a href="#cb8-15" tabindex="-1"></a></span>
<span id="cb8-16"><a href="#cb8-16" tabindex="-1"></a><span class="co"># Perform the likelihood ratio test</span></span>
<span id="cb8-17"><a href="#cb8-17" tabindex="-1"></a>qlf.cbd_healthy_vs_veh_healthy <span class="ot">&lt;-</span> edgeR<span class="sc">::</span><span class="fu">glmQLFTest</span>(</span>
<span id="cb8-18"><a href="#cb8-18" tabindex="-1"></a>    fit,</span>
<span id="cb8-19"><a href="#cb8-19" tabindex="-1"></a>    <span class="at">coef =</span> <span class="st">&#39;groupDMSO&#39;</span></span>
<span id="cb8-20"><a href="#cb8-20" tabindex="-1"></a>)</span>
<span id="cb8-21"><a href="#cb8-21" tabindex="-1"></a></span>
<span id="cb8-22"><a href="#cb8-22" tabindex="-1"></a><span class="co"># Extract the most significant genes</span></span>
<span id="cb8-23"><a href="#cb8-23" tabindex="-1"></a>top_cbd_healthy_vs_veh_healthy <span class="ot">&lt;-</span> edgeR<span class="sc">::</span><span class="fu">topTags</span>(</span>
<span id="cb8-24"><a href="#cb8-24" tabindex="-1"></a>    qlf.cbd_healthy_vs_veh_healthy,</span>
<span id="cb8-25"><a href="#cb8-25" tabindex="-1"></a>    <span class="at">adjust.method =</span> <span class="st">&quot;BH&quot;</span>,</span>
<span id="cb8-26"><a href="#cb8-26" tabindex="-1"></a>    <span class="at">n =</span> <span class="fu">nrow</span>(d)</span>
<span id="cb8-27"><a href="#cb8-27" tabindex="-1"></a>)</span>
<span id="cb8-28"><a href="#cb8-28" tabindex="-1"></a></span>
<span id="cb8-29"><a href="#cb8-29" tabindex="-1"></a><span class="co"># Count the number of significant genes</span></span>
<span id="cb8-30"><a href="#cb8-30" tabindex="-1"></a>n_sig_genes_c1 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb8-31"><a href="#cb8-31" tabindex="-1"></a>    <span class="at">PValue =</span> <span class="fu">length</span>(<span class="fu">which</span>(top_cbd_healthy_vs_veh_healthy<span class="sc">$</span>table<span class="sc">$</span>PValue <span class="sc">&lt;</span> <span class="fl">0.05</span>)),</span>
<span id="cb8-32"><a href="#cb8-32" tabindex="-1"></a>    <span class="at">FDR =</span> <span class="fu">length</span>(<span class="fu">which</span>(top_cbd_healthy_vs_veh_healthy<span class="sc">$</span>table<span class="sc">$</span>FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>))</span>
<span id="cb8-33"><a href="#cb8-33" tabindex="-1"></a>)</span>
<span id="cb8-34"><a href="#cb8-34" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(</span>
<span id="cb8-35"><a href="#cb8-35" tabindex="-1"></a>    n_sig_genes_c1,</span>
<span id="cb8-36"><a href="#cb8-36" tabindex="-1"></a>    <span class="at">type =</span> <span class="st">&quot;html&quot;</span>,</span>
<span id="cb8-37"><a href="#cb8-37" tabindex="-1"></a>    <span class="at">row.names =</span> <span class="cn">TRUE</span>,</span>
<span id="cb8-38"><a href="#cb8-38" tabindex="-1"></a>    <span class="at">caption =</span> <span class="st">&quot;CBD Healthy vs DMSO Healthy&quot;</span></span>
<span id="cb8-39"><a href="#cb8-39" tabindex="-1"></a>)</span></code></pre></div>
<table>
<caption>CBD Healthy vs DMSO Healthy</caption>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">PValue</th>
<th align="right">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">1</td>
<td align="right">11500</td>
<td align="right">10526</td>
</tr>
</tbody>
</table>
</div>
<div id="volcano-plot" class="section level4">
<h4>Volcano Plot</h4>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="co"># Define the genes of interest</span></span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a>genes_of_interest <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;ERN1&quot;</span>, <span class="st">&quot;EIF2AK3&quot;</span>, <span class="st">&quot;ATF6&quot;</span>)</span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a></span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a><span class="co"># Create a data frame of the results</span></span>
<span id="cb9-5"><a href="#cb9-5" tabindex="-1"></a>results <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb9-6"><a href="#cb9-6" tabindex="-1"></a>    <span class="at">Gene =</span> <span class="fu">rownames</span>(top_cbd_healthy_vs_veh_healthy<span class="sc">$</span>table),</span>
<span id="cb9-7"><a href="#cb9-7" tabindex="-1"></a>    <span class="at">logFC =</span> top_cbd_healthy_vs_veh_healthy<span class="sc">$</span>table<span class="sc">$</span>logFC,</span>
<span id="cb9-8"><a href="#cb9-8" tabindex="-1"></a>    <span class="at">PValue =</span> top_cbd_healthy_vs_veh_healthy<span class="sc">$</span>table<span class="sc">$</span>PValue,</span>
<span id="cb9-9"><a href="#cb9-9" tabindex="-1"></a>    <span class="at">FDR =</span> top_cbd_healthy_vs_veh_healthy<span class="sc">$</span>table<span class="sc">$</span>FDR,</span>
<span id="cb9-10"><a href="#cb9-10" tabindex="-1"></a>    <span class="at">Significant =</span> <span class="fu">ifelse</span>(</span>
<span id="cb9-11"><a href="#cb9-11" tabindex="-1"></a>        (top_cbd_healthy_vs_veh_healthy<span class="sc">$</span>table<span class="sc">$</span>FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">&amp;</span> <span class="fu">abs</span>(top_cbd_healthy_vs_veh_healthy<span class="sc">$</span>table<span class="sc">$</span>logFC) <span class="sc">&gt;</span> <span class="dv">1</span>,</span>
<span id="cb9-12"><a href="#cb9-12" tabindex="-1"></a>        <span class="st">&quot;Significant&quot;</span>,</span>
<span id="cb9-13"><a href="#cb9-13" tabindex="-1"></a>        <span class="st">&quot;Not Significant&quot;</span></span>
<span id="cb9-14"><a href="#cb9-14" tabindex="-1"></a>    ),</span>
<span id="cb9-15"><a href="#cb9-15" tabindex="-1"></a>    <span class="at">Interest =</span> <span class="fu">ifelse</span>(</span>
<span id="cb9-16"><a href="#cb9-16" tabindex="-1"></a>        <span class="fu">rownames</span>(top_cbd_healthy_vs_veh_healthy<span class="sc">$</span>table) <span class="sc">%in%</span> genes_of_interest,</span>
<span id="cb9-17"><a href="#cb9-17" tabindex="-1"></a>        <span class="st">&quot;Of Interest&quot;</span>,</span>
<span id="cb9-18"><a href="#cb9-18" tabindex="-1"></a>        <span class="st">&quot;Not of Interest&quot;</span></span>
<span id="cb9-19"><a href="#cb9-19" tabindex="-1"></a>    ),</span>
<span id="cb9-20"><a href="#cb9-20" tabindex="-1"></a>    <span class="at">Regulation =</span> <span class="fu">ifelse</span>(</span>
<span id="cb9-21"><a href="#cb9-21" tabindex="-1"></a>        (top_cbd_healthy_vs_veh_healthy<span class="sc">$</span>table<span class="sc">$</span>logFC <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">&amp;</span> (top_cbd_healthy_vs_veh_healthy<span class="sc">$</span>table<span class="sc">$</span>FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>),</span>
<span id="cb9-22"><a href="#cb9-22" tabindex="-1"></a>        <span class="st">&quot;Upregulated&quot;</span>,</span>
<span id="cb9-23"><a href="#cb9-23" tabindex="-1"></a>        <span class="fu">ifelse</span>(</span>
<span id="cb9-24"><a href="#cb9-24" tabindex="-1"></a>            (top_cbd_healthy_vs_veh_healthy<span class="sc">$</span>table<span class="sc">$</span>logFC <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">1</span>) <span class="sc">&amp;</span> (top_cbd_healthy_vs_veh_healthy<span class="sc">$</span>table<span class="sc">$</span>FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>),</span>
<span id="cb9-25"><a href="#cb9-25" tabindex="-1"></a>            <span class="st">&quot;Downregulated&quot;</span>,</span>
<span id="cb9-26"><a href="#cb9-26" tabindex="-1"></a>            <span class="st">&quot;Not Significant&quot;</span></span>
<span id="cb9-27"><a href="#cb9-27" tabindex="-1"></a>        )</span>
<span id="cb9-28"><a href="#cb9-28" tabindex="-1"></a>    )</span>
<span id="cb9-29"><a href="#cb9-29" tabindex="-1"></a>)</span>
<span id="cb9-30"><a href="#cb9-30" tabindex="-1"></a></span>
<span id="cb9-31"><a href="#cb9-31" tabindex="-1"></a><span class="co"># Plot the volcano plot</span></span>
<span id="cb9-32"><a href="#cb9-32" tabindex="-1"></a>ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(</span>
<span id="cb9-33"><a href="#cb9-33" tabindex="-1"></a>    results,</span>
<span id="cb9-34"><a href="#cb9-34" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">aes</span>(<span class="at">x =</span> logFC, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(FDR), <span class="at">color =</span> Regulation, <span class="at">shape =</span> Interest)</span>
<span id="cb9-35"><a href="#cb9-35" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb9-36"><a href="#cb9-36" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb9-37"><a href="#cb9-37" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(<span class="at">data =</span> <span class="fu">subset</span>(results, Interest <span class="sc">==</span> <span class="st">&quot;Of Interest&quot;</span>),</span>
<span id="cb9-38"><a href="#cb9-38" tabindex="-1"></a>                        ggplot2<span class="sc">::</span><span class="fu">aes</span>(<span class="at">fill =</span> Gene), <span class="at">alpha =</span> <span class="fl">1.0</span>, <span class="at">pch =</span> <span class="dv">21</span>) <span class="sc">+</span></span>
<span id="cb9-39"><a href="#cb9-39" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.05</span>), <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>) <span class="sc">+</span></span>
<span id="cb9-40"><a href="#cb9-40" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>, <span class="dv">1</span>), <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>) <span class="sc">+</span></span>
<span id="cb9-41"><a href="#cb9-41" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb9-42"><a href="#cb9-42" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">&quot;Expression in CBD-treated vs DMSO-treated Healthy A549 Cells&quot;</span>,</span>
<span id="cb9-43"><a href="#cb9-43" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">&quot;Log2 Fold Change&quot;</span>,</span>
<span id="cb9-44"><a href="#cb9-44" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">&quot;-log10 FDR&quot;</span>,</span>
<span id="cb9-45"><a href="#cb9-45" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">&quot;Significance&quot;</span>,</span>
<span id="cb9-46"><a href="#cb9-46" tabindex="-1"></a>        <span class="at">fill =</span> <span class="st">&quot;Gene&quot;</span>,</span>
<span id="cb9-47"><a href="#cb9-47" tabindex="-1"></a>        <span class="at">shape =</span> <span class="st">&quot;Gene of Interest&quot;</span></span>
<span id="cb9-48"><a href="#cb9-48" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb9-49"><a href="#cb9-49" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;Upregulated&quot;</span> <span class="ot">=</span> <span class="st">&quot;red&quot;</span>, <span class="st">&quot;Downregulated&quot;</span> <span class="ot">=</span> <span class="st">&quot;hotpink&quot;</span>, <span class="st">&quot;Not Significant&quot;</span> <span class="ot">=</span> <span class="st">&quot;grey&quot;</span>)) <span class="sc">+</span></span>
<span id="cb9-50"><a href="#cb9-50" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;ERN1&quot;</span> <span class="ot">=</span> <span class="st">&quot;purple&quot;</span>, <span class="st">&quot;EIF2AK3&quot;</span> <span class="ot">=</span> <span class="st">&quot;cyan&quot;</span>, <span class="st">&quot;ATF6&quot;</span> <span class="ot">=</span> <span class="st">&quot;green&quot;</span>)) <span class="sc">+</span></span>
<span id="cb9-51"><a href="#cb9-51" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_shape_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="dv">16</span>, <span class="dv">17</span>)) <span class="sc">+</span></span>
<span id="cb9-52"><a href="#cb9-52" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_minimal</span>()</span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="A2_Notebook_files/figure-html/volcano_healthy-1.png" alt="Volcano Plot of Differential Expression between CBD-treated and DMSO-treated Healthy A549 Cells" width="100%" />
<p class="caption">
Volcano Plot of Differential Expression between CBD-treated and
DMSO-treated Healthy A549 Cells
</p>
</div>
</div>
</div>
<div id="cbd-infection-vs-cbd-healthy" class="section level3">
<h3>CBD Infection vs CBD Healthy</h3>
<div id="gene-counts-1" class="section level4">
<h4>Gene Counts</h4>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a><span class="co"># Subset the data</span></span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a>cbd_data_matrix <span class="ot">&lt;-</span> filtered_data_matrix[, <span class="fu">grepl</span>(<span class="st">&quot;CBD_infect|CBD_mock&quot;</span>, <span class="fu">colnames</span>(filtered_data_matrix))]</span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a></span>
<span id="cb10-4"><a href="#cb10-4" tabindex="-1"></a><span class="co"># Create the DGEList object and the design matrix</span></span>
<span id="cb10-5"><a href="#cb10-5" tabindex="-1"></a>group <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">&quot;Infect&quot;</span>, <span class="dv">3</span>), <span class="fu">rep</span>(<span class="st">&quot;Healthy&quot;</span>, <span class="dv">3</span>)))</span>
<span id="cb10-6"><a href="#cb10-6" tabindex="-1"></a>d <span class="ot">&lt;-</span> edgeR<span class="sc">::</span><span class="fu">DGEList</span>(</span>
<span id="cb10-7"><a href="#cb10-7" tabindex="-1"></a>    <span class="at">counts =</span> <span class="fu">as.matrix</span>(cbd_data_matrix),</span>
<span id="cb10-8"><a href="#cb10-8" tabindex="-1"></a>    <span class="at">group =</span> group</span>
<span id="cb10-9"><a href="#cb10-9" tabindex="-1"></a>)</span>
<span id="cb10-10"><a href="#cb10-10" tabindex="-1"></a>design <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span>group)</span>
<span id="cb10-11"><a href="#cb10-11" tabindex="-1"></a></span>
<span id="cb10-12"><a href="#cb10-12" tabindex="-1"></a><span class="co"># Estimate the dispersion and fit the model</span></span>
<span id="cb10-13"><a href="#cb10-13" tabindex="-1"></a>d <span class="ot">&lt;-</span> edgeR<span class="sc">::</span><span class="fu">estimateDisp</span>(d, design)</span>
<span id="cb10-14"><a href="#cb10-14" tabindex="-1"></a>fit <span class="ot">&lt;-</span> edgeR<span class="sc">::</span><span class="fu">glmQLFit</span>(d, design)</span>
<span id="cb10-15"><a href="#cb10-15" tabindex="-1"></a></span>
<span id="cb10-16"><a href="#cb10-16" tabindex="-1"></a><span class="co"># Perform the likelihood ratio test</span></span>
<span id="cb10-17"><a href="#cb10-17" tabindex="-1"></a>qlf.cbd_infect_vs_cbd_healthy <span class="ot">&lt;-</span> edgeR<span class="sc">::</span><span class="fu">glmQLFTest</span>(</span>
<span id="cb10-18"><a href="#cb10-18" tabindex="-1"></a>    fit,</span>
<span id="cb10-19"><a href="#cb10-19" tabindex="-1"></a>    <span class="at">coef =</span> <span class="st">&#39;groupInfect&#39;</span></span>
<span id="cb10-20"><a href="#cb10-20" tabindex="-1"></a>)</span>
<span id="cb10-21"><a href="#cb10-21" tabindex="-1"></a></span>
<span id="cb10-22"><a href="#cb10-22" tabindex="-1"></a><span class="co"># Extract the most significant genes</span></span>
<span id="cb10-23"><a href="#cb10-23" tabindex="-1"></a>top_cbd_infect_vs_cbd_healthy <span class="ot">&lt;-</span> edgeR<span class="sc">::</span><span class="fu">topTags</span>(</span>
<span id="cb10-24"><a href="#cb10-24" tabindex="-1"></a>    qlf.cbd_infect_vs_cbd_healthy,</span>
<span id="cb10-25"><a href="#cb10-25" tabindex="-1"></a>    <span class="at">adjust.method =</span> <span class="st">&quot;BH&quot;</span>,</span>
<span id="cb10-26"><a href="#cb10-26" tabindex="-1"></a>    <span class="at">n =</span> <span class="fu">nrow</span>(d)</span>
<span id="cb10-27"><a href="#cb10-27" tabindex="-1"></a>)</span>
<span id="cb10-28"><a href="#cb10-28" tabindex="-1"></a></span>
<span id="cb10-29"><a href="#cb10-29" tabindex="-1"></a><span class="co"># Count the number of significant genes</span></span>
<span id="cb10-30"><a href="#cb10-30" tabindex="-1"></a>n_sig_genes_c2 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb10-31"><a href="#cb10-31" tabindex="-1"></a>    <span class="at">PValue =</span> <span class="fu">length</span>(<span class="fu">which</span>(top_cbd_infect_vs_cbd_healthy<span class="sc">$</span>table<span class="sc">$</span>PValue <span class="sc">&lt;</span> <span class="fl">0.05</span>)),</span>
<span id="cb10-32"><a href="#cb10-32" tabindex="-1"></a>    <span class="at">FDR =</span> <span class="fu">length</span>(<span class="fu">which</span>(top_cbd_infect_vs_cbd_healthy<span class="sc">$</span>table<span class="sc">$</span>FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>))</span>
<span id="cb10-33"><a href="#cb10-33" tabindex="-1"></a>)</span>
<span id="cb10-34"><a href="#cb10-34" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(</span>
<span id="cb10-35"><a href="#cb10-35" tabindex="-1"></a>    n_sig_genes_c2,</span>
<span id="cb10-36"><a href="#cb10-36" tabindex="-1"></a>    <span class="at">type =</span> <span class="st">&quot;html&quot;</span>,</span>
<span id="cb10-37"><a href="#cb10-37" tabindex="-1"></a>    <span class="at">row.names =</span> <span class="cn">TRUE</span>,</span>
<span id="cb10-38"><a href="#cb10-38" tabindex="-1"></a>    <span class="at">caption =</span> <span class="st">&quot;CBD Infection vs CBD Healthy&quot;</span></span>
<span id="cb10-39"><a href="#cb10-39" tabindex="-1"></a>)</span></code></pre></div>
<table>
<caption>CBD Infection vs CBD Healthy</caption>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">PValue</th>
<th align="right">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">1</td>
<td align="right">5469</td>
<td align="right">2126</td>
</tr>
</tbody>
</table>
</div>
<div id="volcano-plot-1" class="section level4">
<h4>Volcano Plot</h4>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a><span class="co"># Define the genes of interest</span></span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a>genes_of_interest <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;ERN1&quot;</span>, <span class="st">&quot;EIF2AK3&quot;</span>, <span class="st">&quot;ATF6&quot;</span>)</span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a></span>
<span id="cb11-4"><a href="#cb11-4" tabindex="-1"></a><span class="co"># Create a data frame of the results</span></span>
<span id="cb11-5"><a href="#cb11-5" tabindex="-1"></a>results <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb11-6"><a href="#cb11-6" tabindex="-1"></a>    <span class="at">Gene =</span> <span class="fu">rownames</span>(top_cbd_infect_vs_cbd_healthy<span class="sc">$</span>table),</span>
<span id="cb11-7"><a href="#cb11-7" tabindex="-1"></a>    <span class="at">logFC =</span> top_cbd_infect_vs_cbd_healthy<span class="sc">$</span>table<span class="sc">$</span>logFC,</span>
<span id="cb11-8"><a href="#cb11-8" tabindex="-1"></a>    <span class="at">PValue =</span> top_cbd_infect_vs_cbd_healthy<span class="sc">$</span>table<span class="sc">$</span>PValue,</span>
<span id="cb11-9"><a href="#cb11-9" tabindex="-1"></a>    <span class="at">FDR =</span> top_cbd_infect_vs_cbd_healthy<span class="sc">$</span>table<span class="sc">$</span>FDR,</span>
<span id="cb11-10"><a href="#cb11-10" tabindex="-1"></a>    <span class="at">Significant =</span> <span class="fu">ifelse</span>(</span>
<span id="cb11-11"><a href="#cb11-11" tabindex="-1"></a>        (top_cbd_infect_vs_cbd_healthy<span class="sc">$</span>table<span class="sc">$</span>FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">&amp;</span> <span class="fu">abs</span>(top_cbd_infect_vs_cbd_healthy<span class="sc">$</span>table<span class="sc">$</span>logFC) <span class="sc">&gt;</span> <span class="dv">1</span>,</span>
<span id="cb11-12"><a href="#cb11-12" tabindex="-1"></a>        <span class="st">&quot;Significant&quot;</span>,</span>
<span id="cb11-13"><a href="#cb11-13" tabindex="-1"></a>        <span class="st">&quot;Not Significant&quot;</span></span>
<span id="cb11-14"><a href="#cb11-14" tabindex="-1"></a>    ),</span>
<span id="cb11-15"><a href="#cb11-15" tabindex="-1"></a>    <span class="at">Interest =</span> <span class="fu">ifelse</span>(</span>
<span id="cb11-16"><a href="#cb11-16" tabindex="-1"></a>        <span class="fu">rownames</span>(top_cbd_infect_vs_cbd_healthy<span class="sc">$</span>table) <span class="sc">%in%</span> genes_of_interest,</span>
<span id="cb11-17"><a href="#cb11-17" tabindex="-1"></a>        <span class="st">&quot;Of Interest&quot;</span>,</span>
<span id="cb11-18"><a href="#cb11-18" tabindex="-1"></a>        <span class="st">&quot;Not of Interest&quot;</span></span>
<span id="cb11-19"><a href="#cb11-19" tabindex="-1"></a>    ),</span>
<span id="cb11-20"><a href="#cb11-20" tabindex="-1"></a>    <span class="at">Regulation =</span> <span class="fu">ifelse</span>(</span>
<span id="cb11-21"><a href="#cb11-21" tabindex="-1"></a>        (top_cbd_infect_vs_cbd_healthy<span class="sc">$</span>table<span class="sc">$</span>logFC <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">&amp;</span> (top_cbd_infect_vs_cbd_healthy<span class="sc">$</span>table<span class="sc">$</span>FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>),</span>
<span id="cb11-22"><a href="#cb11-22" tabindex="-1"></a>        <span class="st">&quot;Upregulated&quot;</span>,</span>
<span id="cb11-23"><a href="#cb11-23" tabindex="-1"></a>        <span class="fu">ifelse</span>(</span>
<span id="cb11-24"><a href="#cb11-24" tabindex="-1"></a>            (top_cbd_infect_vs_cbd_healthy<span class="sc">$</span>table<span class="sc">$</span>logFC <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">1</span>) <span class="sc">&amp;</span> (top_cbd_infect_vs_cbd_healthy<span class="sc">$</span>table<span class="sc">$</span>FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>),</span>
<span id="cb11-25"><a href="#cb11-25" tabindex="-1"></a>            <span class="st">&quot;Downregulated&quot;</span>,</span>
<span id="cb11-26"><a href="#cb11-26" tabindex="-1"></a>            <span class="st">&quot;Not Significant&quot;</span></span>
<span id="cb11-27"><a href="#cb11-27" tabindex="-1"></a>        )</span>
<span id="cb11-28"><a href="#cb11-28" tabindex="-1"></a>    )</span>
<span id="cb11-29"><a href="#cb11-29" tabindex="-1"></a>)</span>
<span id="cb11-30"><a href="#cb11-30" tabindex="-1"></a></span>
<span id="cb11-31"><a href="#cb11-31" tabindex="-1"></a><span class="co"># Plot the volcano plot</span></span>
<span id="cb11-32"><a href="#cb11-32" tabindex="-1"></a>ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(</span>
<span id="cb11-33"><a href="#cb11-33" tabindex="-1"></a>    results,</span>
<span id="cb11-34"><a href="#cb11-34" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">aes</span>(<span class="at">x =</span> logFC, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(FDR), <span class="at">color =</span> Regulation, <span class="at">shape =</span> Interest)</span>
<span id="cb11-35"><a href="#cb11-35" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb11-36"><a href="#cb11-36" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb11-37"><a href="#cb11-37" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(<span class="at">data =</span> <span class="fu">subset</span>(results, Interest <span class="sc">==</span> <span class="st">&quot;Of Interest&quot;</span>),</span>
<span id="cb11-38"><a href="#cb11-38" tabindex="-1"></a>                        ggplot2<span class="sc">::</span><span class="fu">aes</span>(<span class="at">fill =</span> Gene), <span class="at">alpha =</span> <span class="fl">1.0</span>, <span class="at">pch =</span> <span class="dv">21</span>) <span class="sc">+</span></span>
<span id="cb11-39"><a href="#cb11-39" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.05</span>), <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>) <span class="sc">+</span></span>
<span id="cb11-40"><a href="#cb11-40" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>, <span class="dv">1</span>), <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>) <span class="sc">+</span></span>
<span id="cb11-41"><a href="#cb11-41" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb11-42"><a href="#cb11-42" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">&quot;Expression in Healthy vs. Infected A549 Cells Treated with CBD&quot;</span>,</span>
<span id="cb11-43"><a href="#cb11-43" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">&quot;Log2 Fold Change&quot;</span>,</span>
<span id="cb11-44"><a href="#cb11-44" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">&quot;-log10 FDR&quot;</span>,</span>
<span id="cb11-45"><a href="#cb11-45" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">&quot;Significance&quot;</span>,</span>
<span id="cb11-46"><a href="#cb11-46" tabindex="-1"></a>        <span class="at">fill =</span> <span class="st">&quot;Gene&quot;</span>,</span>
<span id="cb11-47"><a href="#cb11-47" tabindex="-1"></a>        <span class="at">shape =</span> <span class="st">&quot;Gene of Interest&quot;</span></span>
<span id="cb11-48"><a href="#cb11-48" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb11-49"><a href="#cb11-49" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;Upregulated&quot;</span> <span class="ot">=</span> <span class="st">&quot;red&quot;</span>, <span class="st">&quot;Downregulated&quot;</span> <span class="ot">=</span> <span class="st">&quot;hotpink&quot;</span>, <span class="st">&quot;Not Significant&quot;</span> <span class="ot">=</span> <span class="st">&quot;grey&quot;</span>)) <span class="sc">+</span></span>
<span id="cb11-50"><a href="#cb11-50" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;ERN1&quot;</span> <span class="ot">=</span> <span class="st">&quot;purple&quot;</span>, <span class="st">&quot;EIF2AK3&quot;</span> <span class="ot">=</span> <span class="st">&quot;cyan&quot;</span>, <span class="st">&quot;ATF6&quot;</span> <span class="ot">=</span> <span class="st">&quot;green&quot;</span>)) <span class="sc">+</span></span>
<span id="cb11-51"><a href="#cb11-51" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_shape_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="dv">16</span>, <span class="dv">17</span>)) <span class="sc">+</span></span>
<span id="cb11-52"><a href="#cb11-52" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_minimal</span>()</span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="A2_Notebook_files/figure-html/volcano_cbd-1.png" alt="Volcano Plot of Differential Expression between Healthy vs. Infected A549 Cells Treated with CBD" width="100%" />
<p class="caption">
Volcano Plot of Differential Expression between Healthy vs. Infected
A549 Cells Treated with CBD
</p>
</div>
</div>
</div>
<div id="cbd-infection-vs-dmso-infection" class="section level3">
<h3>CBD Infection vs DMSO Infection</h3>
<div id="gene-counts-2" class="section level4">
<h4>Gene Counts</h4>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a><span class="co"># Subset the data</span></span>
<span id="cb12-2"><a href="#cb12-2" tabindex="-1"></a>infect_data_matrix <span class="ot">&lt;-</span> filtered_data_matrix[, <span class="fu">grepl</span>(<span class="st">&quot;Veh_infect|CBD_infect&quot;</span>, <span class="fu">colnames</span>(filtered_data_matrix))]</span>
<span id="cb12-3"><a href="#cb12-3" tabindex="-1"></a>group <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">&quot;CBD&quot;</span>, <span class="dv">3</span>), <span class="fu">rep</span>(<span class="st">&quot;DMSO&quot;</span>, <span class="dv">3</span>)))</span>
<span id="cb12-4"><a href="#cb12-4" tabindex="-1"></a></span>
<span id="cb12-5"><a href="#cb12-5" tabindex="-1"></a><span class="co"># Create the DGEList object and the design matrix</span></span>
<span id="cb12-6"><a href="#cb12-6" tabindex="-1"></a>d <span class="ot">&lt;-</span> edgeR<span class="sc">::</span><span class="fu">DGEList</span>(</span>
<span id="cb12-7"><a href="#cb12-7" tabindex="-1"></a>    <span class="at">counts =</span> <span class="fu">as.matrix</span>(infect_data_matrix),</span>
<span id="cb12-8"><a href="#cb12-8" tabindex="-1"></a>    <span class="at">group =</span> group</span>
<span id="cb12-9"><a href="#cb12-9" tabindex="-1"></a>)</span>
<span id="cb12-10"><a href="#cb12-10" tabindex="-1"></a>design <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span>group)</span>
<span id="cb12-11"><a href="#cb12-11" tabindex="-1"></a></span>
<span id="cb12-12"><a href="#cb12-12" tabindex="-1"></a><span class="co"># Estimate the dispersion and fit the model</span></span>
<span id="cb12-13"><a href="#cb12-13" tabindex="-1"></a>d <span class="ot">&lt;-</span> edgeR<span class="sc">::</span><span class="fu">estimateDisp</span>(d, design)</span>
<span id="cb12-14"><a href="#cb12-14" tabindex="-1"></a>fit <span class="ot">&lt;-</span> edgeR<span class="sc">::</span><span class="fu">glmQLFit</span>(d, design)</span>
<span id="cb12-15"><a href="#cb12-15" tabindex="-1"></a></span>
<span id="cb12-16"><a href="#cb12-16" tabindex="-1"></a><span class="co"># Perform the likelihood ratio test</span></span>
<span id="cb12-17"><a href="#cb12-17" tabindex="-1"></a>qlf.cbd_infect_vs_veh_infect <span class="ot">&lt;-</span> edgeR<span class="sc">::</span><span class="fu">glmQLFTest</span>(</span>
<span id="cb12-18"><a href="#cb12-18" tabindex="-1"></a>    fit,</span>
<span id="cb12-19"><a href="#cb12-19" tabindex="-1"></a>    <span class="at">coef =</span> <span class="st">&#39;groupDMSO&#39;</span></span>
<span id="cb12-20"><a href="#cb12-20" tabindex="-1"></a>)</span>
<span id="cb12-21"><a href="#cb12-21" tabindex="-1"></a></span>
<span id="cb12-22"><a href="#cb12-22" tabindex="-1"></a><span class="co"># Extract the most significant genes</span></span>
<span id="cb12-23"><a href="#cb12-23" tabindex="-1"></a>top_cbd_infect_vs_veh_infect <span class="ot">&lt;-</span> edgeR<span class="sc">::</span><span class="fu">topTags</span>(</span>
<span id="cb12-24"><a href="#cb12-24" tabindex="-1"></a>    qlf.cbd_infect_vs_veh_infect,</span>
<span id="cb12-25"><a href="#cb12-25" tabindex="-1"></a>    <span class="at">adjust.method =</span> <span class="st">&quot;BH&quot;</span>,</span>
<span id="cb12-26"><a href="#cb12-26" tabindex="-1"></a>    <span class="at">n =</span> <span class="fu">nrow</span>(d)</span>
<span id="cb12-27"><a href="#cb12-27" tabindex="-1"></a>)</span>
<span id="cb12-28"><a href="#cb12-28" tabindex="-1"></a></span>
<span id="cb12-29"><a href="#cb12-29" tabindex="-1"></a><span class="co"># Count the number of significant genes</span></span>
<span id="cb12-30"><a href="#cb12-30" tabindex="-1"></a>n_sig_genes_c3 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb12-31"><a href="#cb12-31" tabindex="-1"></a>    <span class="at">PValue =</span> <span class="fu">length</span>(<span class="fu">which</span>(top_cbd_infect_vs_veh_infect<span class="sc">$</span>table<span class="sc">$</span>PValue <span class="sc">&lt;</span> <span class="fl">0.05</span>)),</span>
<span id="cb12-32"><a href="#cb12-32" tabindex="-1"></a>    <span class="at">FDR =</span> <span class="fu">length</span>(<span class="fu">which</span>(top_cbd_infect_vs_veh_infect<span class="sc">$</span>table<span class="sc">$</span>FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>))</span>
<span id="cb12-33"><a href="#cb12-33" tabindex="-1"></a>)</span>
<span id="cb12-34"><a href="#cb12-34" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(</span>
<span id="cb12-35"><a href="#cb12-35" tabindex="-1"></a>    n_sig_genes_c3,</span>
<span id="cb12-36"><a href="#cb12-36" tabindex="-1"></a>    <span class="at">type =</span> <span class="st">&quot;html&quot;</span>,</span>
<span id="cb12-37"><a href="#cb12-37" tabindex="-1"></a>    <span class="at">row.names =</span> <span class="cn">TRUE</span>,</span>
<span id="cb12-38"><a href="#cb12-38" tabindex="-1"></a>    <span class="at">caption =</span> <span class="st">&quot;CBD Infection vs DMSO Infection&quot;</span></span>
<span id="cb12-39"><a href="#cb12-39" tabindex="-1"></a>)</span></code></pre></div>
<table>
<caption>CBD Infection vs DMSO Infection</caption>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">PValue</th>
<th align="right">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">1</td>
<td align="right">12228</td>
<td align="right">11353</td>
</tr>
</tbody>
</table>
</div>
<div id="volcano-plot-2" class="section level4">
<h4>Volcano Plot</h4>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a><span class="co"># Define the genes of interest</span></span>
<span id="cb13-2"><a href="#cb13-2" tabindex="-1"></a>genes_of_interest <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;ERN1&quot;</span>, <span class="st">&quot;EIF2AK3&quot;</span>, <span class="st">&quot;ATF6&quot;</span>)</span>
<span id="cb13-3"><a href="#cb13-3" tabindex="-1"></a></span>
<span id="cb13-4"><a href="#cb13-4" tabindex="-1"></a><span class="co"># Create a data frame of the results</span></span>
<span id="cb13-5"><a href="#cb13-5" tabindex="-1"></a>results <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb13-6"><a href="#cb13-6" tabindex="-1"></a>    <span class="at">Gene =</span> <span class="fu">rownames</span>(top_cbd_infect_vs_veh_infect<span class="sc">$</span>table),</span>
<span id="cb13-7"><a href="#cb13-7" tabindex="-1"></a>    <span class="at">logFC =</span> top_cbd_infect_vs_veh_infect<span class="sc">$</span>table<span class="sc">$</span>logFC,</span>
<span id="cb13-8"><a href="#cb13-8" tabindex="-1"></a>    <span class="at">PValue =</span> top_cbd_infect_vs_veh_infect<span class="sc">$</span>table<span class="sc">$</span>PValue,</span>
<span id="cb13-9"><a href="#cb13-9" tabindex="-1"></a>    <span class="at">FDR =</span> top_cbd_infect_vs_veh_infect<span class="sc">$</span>table<span class="sc">$</span>FDR,</span>
<span id="cb13-10"><a href="#cb13-10" tabindex="-1"></a>    <span class="at">Significant =</span> <span class="fu">ifelse</span>(</span>
<span id="cb13-11"><a href="#cb13-11" tabindex="-1"></a>        (top_cbd_infect_vs_veh_infect<span class="sc">$</span>table<span class="sc">$</span>FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">&amp;</span> <span class="fu">abs</span>(top_cbd_infect_vs_veh_infect<span class="sc">$</span>table<span class="sc">$</span>logFC) <span class="sc">&gt;</span> <span class="dv">1</span>,</span>
<span id="cb13-12"><a href="#cb13-12" tabindex="-1"></a>        <span class="st">&quot;Significant&quot;</span>,</span>
<span id="cb13-13"><a href="#cb13-13" tabindex="-1"></a>        <span class="st">&quot;Not Significant&quot;</span></span>
<span id="cb13-14"><a href="#cb13-14" tabindex="-1"></a>    ),</span>
<span id="cb13-15"><a href="#cb13-15" tabindex="-1"></a>    <span class="at">Interest =</span> <span class="fu">ifelse</span>(</span>
<span id="cb13-16"><a href="#cb13-16" tabindex="-1"></a>        <span class="fu">rownames</span>(top_cbd_infect_vs_veh_infect<span class="sc">$</span>table) <span class="sc">%in%</span> genes_of_interest,</span>
<span id="cb13-17"><a href="#cb13-17" tabindex="-1"></a>        <span class="st">&quot;Of Interest&quot;</span>,</span>
<span id="cb13-18"><a href="#cb13-18" tabindex="-1"></a>        <span class="st">&quot;Not of Interest&quot;</span></span>
<span id="cb13-19"><a href="#cb13-19" tabindex="-1"></a>    ),</span>
<span id="cb13-20"><a href="#cb13-20" tabindex="-1"></a>    <span class="at">Regulation =</span> <span class="fu">ifelse</span>(</span>
<span id="cb13-21"><a href="#cb13-21" tabindex="-1"></a>        (top_cbd_infect_vs_veh_infect<span class="sc">$</span>table<span class="sc">$</span>logFC <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">&amp;</span> (top_cbd_infect_vs_veh_infect<span class="sc">$</span>table<span class="sc">$</span>FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>),</span>
<span id="cb13-22"><a href="#cb13-22" tabindex="-1"></a>        <span class="st">&quot;Upregulated&quot;</span>,</span>
<span id="cb13-23"><a href="#cb13-23" tabindex="-1"></a>        <span class="fu">ifelse</span>(</span>
<span id="cb13-24"><a href="#cb13-24" tabindex="-1"></a>            (top_cbd_infect_vs_veh_infect<span class="sc">$</span>table<span class="sc">$</span>logFC <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">1</span>) <span class="sc">&amp;</span> (top_cbd_infect_vs_veh_infect<span class="sc">$</span>table<span class="sc">$</span>FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>),</span>
<span id="cb13-25"><a href="#cb13-25" tabindex="-1"></a>            <span class="st">&quot;Downregulated&quot;</span>,</span>
<span id="cb13-26"><a href="#cb13-26" tabindex="-1"></a>            <span class="st">&quot;Not Significant&quot;</span></span>
<span id="cb13-27"><a href="#cb13-27" tabindex="-1"></a>        )</span>
<span id="cb13-28"><a href="#cb13-28" tabindex="-1"></a>    )</span>
<span id="cb13-29"><a href="#cb13-29" tabindex="-1"></a>)</span>
<span id="cb13-30"><a href="#cb13-30" tabindex="-1"></a></span>
<span id="cb13-31"><a href="#cb13-31" tabindex="-1"></a><span class="co"># Plot the volcano plot</span></span>
<span id="cb13-32"><a href="#cb13-32" tabindex="-1"></a>ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(</span>
<span id="cb13-33"><a href="#cb13-33" tabindex="-1"></a>    results,</span>
<span id="cb13-34"><a href="#cb13-34" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">aes</span>(<span class="at">x =</span> logFC, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(FDR), <span class="at">color =</span> Regulation, <span class="at">shape =</span> Interest)</span>
<span id="cb13-35"><a href="#cb13-35" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb13-36"><a href="#cb13-36" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb13-37"><a href="#cb13-37" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(<span class="at">data =</span> <span class="fu">subset</span>(results, Interest <span class="sc">==</span> <span class="st">&quot;Of Interest&quot;</span>),</span>
<span id="cb13-38"><a href="#cb13-38" tabindex="-1"></a>                        ggplot2<span class="sc">::</span><span class="fu">aes</span>(<span class="at">fill =</span> Gene), <span class="at">alpha =</span> <span class="fl">1.0</span>, <span class="at">pch =</span> <span class="dv">21</span>) <span class="sc">+</span></span>
<span id="cb13-39"><a href="#cb13-39" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.05</span>), <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>) <span class="sc">+</span></span>
<span id="cb13-40"><a href="#cb13-40" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>, <span class="dv">1</span>), <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>) <span class="sc">+</span></span>
<span id="cb13-41"><a href="#cb13-41" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb13-42"><a href="#cb13-42" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">&quot;Expression in CBD-treated vs DMSO-treated Infected A549 Cells&quot;</span>,</span>
<span id="cb13-43"><a href="#cb13-43" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">&quot;Log2 Fold Change&quot;</span>,</span>
<span id="cb13-44"><a href="#cb13-44" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">&quot;-log10 FDR&quot;</span>,</span>
<span id="cb13-45"><a href="#cb13-45" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">&quot;Significance&quot;</span>,</span>
<span id="cb13-46"><a href="#cb13-46" tabindex="-1"></a>        <span class="at">fill =</span> <span class="st">&quot;Gene&quot;</span>,</span>
<span id="cb13-47"><a href="#cb13-47" tabindex="-1"></a>        <span class="at">shape =</span> <span class="st">&quot;Gene of Interest&quot;</span></span>
<span id="cb13-48"><a href="#cb13-48" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb13-49"><a href="#cb13-49" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;Upregulated&quot;</span> <span class="ot">=</span> <span class="st">&quot;red&quot;</span>, <span class="st">&quot;Downregulated&quot;</span> <span class="ot">=</span> <span class="st">&quot;hotpink&quot;</span>, <span class="st">&quot;Not Significant&quot;</span> <span class="ot">=</span> <span class="st">&quot;grey&quot;</span>)) <span class="sc">+</span></span>
<span id="cb13-50"><a href="#cb13-50" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;ERN1&quot;</span> <span class="ot">=</span> <span class="st">&quot;purple&quot;</span>, <span class="st">&quot;EIF2AK3&quot;</span> <span class="ot">=</span> <span class="st">&quot;cyan&quot;</span>, <span class="st">&quot;ATF6&quot;</span> <span class="ot">=</span> <span class="st">&quot;green&quot;</span>)) <span class="sc">+</span></span>
<span id="cb13-51"><a href="#cb13-51" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_shape_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="dv">16</span>, <span class="dv">17</span>)) <span class="sc">+</span></span>
<span id="cb13-52"><a href="#cb13-52" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_minimal</span>()</span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="A2_Notebook_files/figure-html/volcano_infect-1.png" alt="Volcano Plot of Differential Expression between CBD-treated and DMSO-treated Infected A549 Cells" width="100%" />
<p class="caption">
Volcano Plot of Differential Expression between CBD-treated and
DMSO-treated Infected A549 Cells
</p>
</div>
</div>
</div>
</div>
<div id="further-analysis-of-differential-expression"
class="section level2">
<h2>Further Analysis of Differential Expression</h2>
<p>From the conditions examined above, we will proceed with the
<strong>CBD Infection vs CBD Healthy</strong> condition for further
analysis.</p>
<div id="heatmap" class="section level3">
<h3>Heatmap</h3>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a><span class="co"># Subset the data</span></span>
<span id="cb14-2"><a href="#cb14-2" tabindex="-1"></a>cbd_data_matrix <span class="ot">&lt;-</span> filtered_data_matrix[, <span class="fu">grepl</span>(<span class="st">&quot;CBD_infect|CBD_mock&quot;</span>, <span class="fu">colnames</span>(filtered_data_matrix))]</span>
<span id="cb14-3"><a href="#cb14-3" tabindex="-1"></a></span>
<span id="cb14-4"><a href="#cb14-4" tabindex="-1"></a><span class="co"># Create a numerical matrix of our count data</span></span>
<span id="cb14-5"><a href="#cb14-5" tabindex="-1"></a>heatmap_matrix <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(cbd_data_matrix)</span>
<span id="cb14-6"><a href="#cb14-6" tabindex="-1"></a></span>
<span id="cb14-7"><a href="#cb14-7" tabindex="-1"></a><span class="co"># Set the row names to be the gene names</span></span>
<span id="cb14-8"><a href="#cb14-8" tabindex="-1"></a><span class="fu">rownames</span>(heatmap_matrix) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(cbd_data_matrix)</span>
<span id="cb14-9"><a href="#cb14-9" tabindex="-1"></a></span>
<span id="cb14-10"><a href="#cb14-10" tabindex="-1"></a><span class="co"># Set the column names to be the sample names</span></span>
<span id="cb14-11"><a href="#cb14-11" tabindex="-1"></a><span class="fu">colnames</span>(heatmap_matrix) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;CBD + SARS-CoV-2&quot;</span>, <span class="st">&quot;CBD + SARS-CoV-2&quot;</span>, <span class="st">&quot;CBD + SARS-CoV-2&quot;</span>, <span class="st">&quot;CBD + Mock&quot;</span>, <span class="st">&quot;CBD + Mock&quot;</span>, <span class="st">&quot;CBD + Mock&quot;</span>)</span>
<span id="cb14-12"><a href="#cb14-12" tabindex="-1"></a></span>
<span id="cb14-13"><a href="#cb14-13" tabindex="-1"></a><span class="co"># Get the row names of the significantly differentially expressed genes</span></span>
<span id="cb14-14"><a href="#cb14-14" tabindex="-1"></a>sig_genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(qlf.cbd_infect_vs_cbd_healthy<span class="sc">$</span>table[top_cbd_infect_vs_cbd_healthy<span class="sc">$</span>table<span class="sc">$</span>FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>, ])</span>
<span id="cb14-15"><a href="#cb14-15" tabindex="-1"></a></span>
<span id="cb14-16"><a href="#cb14-16" tabindex="-1"></a><span class="co"># Subset the matrix to only include the significantly differentially expressed genes</span></span>
<span id="cb14-17"><a href="#cb14-17" tabindex="-1"></a>heatmap_matrix <span class="ot">&lt;-</span> heatmap_matrix[sig_genes, ]</span>
<span id="cb14-18"><a href="#cb14-18" tabindex="-1"></a></span>
<span id="cb14-19"><a href="#cb14-19" tabindex="-1"></a><span class="co"># Scale the matrix</span></span>
<span id="cb14-20"><a href="#cb14-20" tabindex="-1"></a>heatmap_matrix <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(heatmap_matrix)))</span>
<span id="cb14-21"><a href="#cb14-21" tabindex="-1"></a></span>
<span id="cb14-22"><a href="#cb14-22" tabindex="-1"></a><span class="co"># Check if the matrix has negative values</span></span>
<span id="cb14-23"><a href="#cb14-23" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">min</span>(heatmap_matrix) <span class="sc">&lt;</span> <span class="dv">0</span>) {</span>
<span id="cb14-24"><a href="#cb14-24" tabindex="-1"></a>    heatmap_col <span class="ot">&lt;-</span> circlize<span class="sc">::</span><span class="fu">colorRamp2</span>(</span>
<span id="cb14-25"><a href="#cb14-25" tabindex="-1"></a>        <span class="fu">c</span>(<span class="fu">min</span>(heatmap_matrix), <span class="dv">0</span>, <span class="fu">max</span>(heatmap_matrix)),</span>
<span id="cb14-26"><a href="#cb14-26" tabindex="-1"></a>        <span class="fu">c</span>(<span class="st">&quot;blue&quot;</span>, <span class="st">&quot;white&quot;</span>, <span class="st">&quot;red&quot;</span>)</span>
<span id="cb14-27"><a href="#cb14-27" tabindex="-1"></a>    )</span>
<span id="cb14-28"><a href="#cb14-28" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb14-29"><a href="#cb14-29" tabindex="-1"></a>    heatmap_col <span class="ot">&lt;-</span> circlize<span class="sc">::</span><span class="fu">colorRamp2</span>(</span>
<span id="cb14-30"><a href="#cb14-30" tabindex="-1"></a>        <span class="fu">range</span>(heatmap_matrix),</span>
<span id="cb14-31"><a href="#cb14-31" tabindex="-1"></a>        <span class="fu">c</span>(<span class="st">&quot;white&quot;</span>, <span class="st">&quot;red&quot;</span>)</span>
<span id="cb14-32"><a href="#cb14-32" tabindex="-1"></a>    )</span>
<span id="cb14-33"><a href="#cb14-33" tabindex="-1"></a>}</span>
<span id="cb14-34"><a href="#cb14-34" tabindex="-1"></a></span>
<span id="cb14-35"><a href="#cb14-35" tabindex="-1"></a><span class="co"># Create an annotation data frame</span></span>
<span id="cb14-36"><a href="#cb14-36" tabindex="-1"></a>annotation_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb14-37"><a href="#cb14-37" tabindex="-1"></a>    <span class="at">CBD =</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">&quot;+&quot;</span>, <span class="dv">6</span>)),</span>
<span id="cb14-38"><a href="#cb14-38" tabindex="-1"></a>    <span class="st">`</span><span class="at">SARS-CoV-2</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">&quot;+&quot;</span>, <span class="dv">3</span>), <span class="fu">rep</span>(<span class="st">&quot;-&quot;</span>, <span class="dv">3</span>)),</span>
<span id="cb14-39"><a href="#cb14-39" tabindex="-1"></a>    <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span></span>
<span id="cb14-40"><a href="#cb14-40" tabindex="-1"></a>)</span>
<span id="cb14-41"><a href="#cb14-41" tabindex="-1"></a>annotation_colours <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb14-42"><a href="#cb14-42" tabindex="-1"></a>    <span class="at">CBD =</span> <span class="fu">c</span>(<span class="st">&quot;+&quot;</span> <span class="ot">=</span> <span class="st">&quot;purple&quot;</span>),</span>
<span id="cb14-43"><a href="#cb14-43" tabindex="-1"></a>    <span class="st">`</span><span class="at">SARS-CoV-2</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;+&quot;</span> <span class="ot">=</span> <span class="st">&quot;green&quot;</span>, <span class="st">&quot;-&quot;</span> <span class="ot">=</span> <span class="st">&quot;yellow&quot;</span>)</span>
<span id="cb14-44"><a href="#cb14-44" tabindex="-1"></a>)</span>
<span id="cb14-45"><a href="#cb14-45" tabindex="-1"></a></span>
<span id="cb14-46"><a href="#cb14-46" tabindex="-1"></a><span class="co"># Add the annotation</span></span>
<span id="cb14-47"><a href="#cb14-47" tabindex="-1"></a>heatmap_annotation <span class="ot">&lt;-</span> ComplexHeatmap<span class="sc">::</span><span class="fu">HeatmapAnnotation</span>(</span>
<span id="cb14-48"><a href="#cb14-48" tabindex="-1"></a>    <span class="at">df =</span> annotation_df,</span>
<span id="cb14-49"><a href="#cb14-49" tabindex="-1"></a>    <span class="at">col =</span> annotation_colours</span>
<span id="cb14-50"><a href="#cb14-50" tabindex="-1"></a>)</span>
<span id="cb14-51"><a href="#cb14-51" tabindex="-1"></a></span>
<span id="cb14-52"><a href="#cb14-52" tabindex="-1"></a><span class="co"># Create the heatmap</span></span>
<span id="cb14-53"><a href="#cb14-53" tabindex="-1"></a>ComplexHeatmap<span class="sc">::</span><span class="fu">Heatmap</span>(</span>
<span id="cb14-54"><a href="#cb14-54" tabindex="-1"></a>    <span class="fu">as.matrix</span>(heatmap_matrix),</span>
<span id="cb14-55"><a href="#cb14-55" tabindex="-1"></a>    <span class="at">col =</span> heatmap_col,</span>
<span id="cb14-56"><a href="#cb14-56" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">&quot;Log2 TPM&quot;</span>,</span>
<span id="cb14-57"><a href="#cb14-57" tabindex="-1"></a>    <span class="at">cluster_rows =</span> <span class="cn">TRUE</span>,</span>
<span id="cb14-58"><a href="#cb14-58" tabindex="-1"></a>    <span class="at">cluster_columns =</span> <span class="cn">TRUE</span>,</span>
<span id="cb14-59"><a href="#cb14-59" tabindex="-1"></a>    <span class="at">show_row_dend =</span> <span class="cn">TRUE</span>,</span>
<span id="cb14-60"><a href="#cb14-60" tabindex="-1"></a>    <span class="at">show_column_dend =</span> <span class="cn">TRUE</span>,</span>
<span id="cb14-61"><a href="#cb14-61" tabindex="-1"></a>    <span class="at">show_row_names =</span> <span class="cn">FALSE</span>,</span>
<span id="cb14-62"><a href="#cb14-62" tabindex="-1"></a>    <span class="at">show_column_names =</span> <span class="cn">FALSE</span>,</span>
<span id="cb14-63"><a href="#cb14-63" tabindex="-1"></a>    <span class="at">show_heatmap_legend =</span> <span class="cn">TRUE</span>,</span>
<span id="cb14-64"><a href="#cb14-64" tabindex="-1"></a>    <span class="at">top_annotation =</span> heatmap_annotation</span>
<span id="cb14-65"><a href="#cb14-65" tabindex="-1"></a>)</span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="A2_Notebook_files/figure-html/heatmap-1.png" alt="Heatmap of Significantly Differentially Expressed Genes" width="100%" />
<p class="caption">
Heatmap of Significantly Differentially Expressed Genes
</p>
</div>
</div>
<div id="questions" class="section level3">
<h3>Questions</h3>
<p><em>The questions below are based on the analysis of the differential
expression data comparing the CBD Infection vs CBD Healthy
conditions.</em></p>
<p><strong>1. How many genes are significantly differentially expressed?
What thresholds did you use and why?</strong></p>
<p>We used quasi-likelihood (QL) to fit the model and perform the
likelihood ratio test. Through the QLF test, we found 5469 genes were
significantly expressed, with a p-value threshold of 0.05. Additionally,
we also observed that 2126 genes passed the False Discovery Rate (FDR)
threshold of 0.05.</p>
<p><em>FDR Reasoning</em>: The FDR threshold of 0.05 was selected
because initial analysis using 0.01 (as mentioned in the paper) excluded
many genes and led to a heatmap where clusters were not as clear. The
0.05 threshold allowed for a more comprehensive analysis of the
data.</p>
<p><strong>2. Which method did you use to adjust for multiple testing
and why?</strong></p>
<p>We used the Benjamini-Hochberg (BH) method to adjust for multiple
testing. It is the method discussed in-depth in the lecture material, as
well as the method used by the authors in the original work with this
dataset (although they don’t mention the BH method explicitly, they
mention the FDR method which is the same as the BH method)<span
class="citation"><sup>4</sup></span>.</p>
<p><strong>3. How many genes passed correction? Are any of them genes of
interest?</strong></p>
<p>2126 genes passed the FDR correction.</p>
<p>In the publication, Nguyen et al. (2022) specifically discuss the
effects of CBD treatment on the endoplasmic reticulum (ER) stress
response - a response mediated by IRE1α, PERK, and ATF6. They found that
CBD treatment will induce the ER stress response<span
class="citation"><sup>4</sup></span>. These genes are highlighted in all
three volcano plots above (CBD Healthy vs DMSO Healthy, CBD Infection vs
CBD Healthy, and CBD Infection vs DMSO Infection).</p>
<p><strong>4. Do your conditions cluster together in the
heatmap?</strong></p>
<p>The heatmap shows that the samples from the same condition cluster
together, indicating that the gene expression profiles are similar
within each condition. This clustering indicates that the gene
expression profiles are similar within each condition, despite the
samples being different.</p>
</div>
</div>
</div>
<div id="thresholded-over-representation-analysis"
class="section level1">
<h1>Thresholded Over-representation Analysis</h1>
<p>Now that we have identified the significantly differentially
expressed genes, we will perform a thresholded over-representation
analysis (ORA) to identify the biological processes that are enriched in
the differentially expressed genes. To do this, we will use
<code>g:Profiler</code> to perform the ORA.</p>
<div id="classifying-upregulated-and-downregulated-genes"
class="section level2">
<h2>Classifying Upregulated and Downregulated Genes</h2>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a><span class="co"># Get which genes are significantly upregulated</span></span>
<span id="cb15-2"><a href="#cb15-2" tabindex="-1"></a>upregulated_genes <span class="ot">&lt;-</span> <span class="fu">which</span>(top_cbd_infect_vs_cbd_healthy<span class="sc">$</span>table<span class="sc">$</span>logFC <span class="sc">&gt;</span> <span class="dv">0</span> <span class="sc">&amp;</span> top_cbd_infect_vs_cbd_healthy<span class="sc">$</span>table<span class="sc">$</span>FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb15-3"><a href="#cb15-3" tabindex="-1"></a>upregulated_genes <span class="ot">&lt;-</span> top_cbd_infect_vs_cbd_healthy[upregulated_genes,][, <span class="fu">c</span>(<span class="st">&quot;logFC&quot;</span>, <span class="st">&quot;PValue&quot;</span>, <span class="st">&quot;FDR&quot;</span>)]</span>
<span id="cb15-4"><a href="#cb15-4" tabindex="-1"></a></span>
<span id="cb15-5"><a href="#cb15-5" tabindex="-1"></a><span class="co"># Create a data frame of the upregulated genes</span></span>
<span id="cb15-6"><a href="#cb15-6" tabindex="-1"></a>upregulated_genes <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb15-7"><a href="#cb15-7" tabindex="-1"></a>    upregulated_genes,</span>
<span id="cb15-8"><a href="#cb15-8" tabindex="-1"></a>    <span class="at">row.names =</span> <span class="fu">rownames</span>(upregulated_genes)</span>
<span id="cb15-9"><a href="#cb15-9" tabindex="-1"></a>)</span>
<span id="cb15-10"><a href="#cb15-10" tabindex="-1"></a></span>
<span id="cb15-11"><a href="#cb15-11" tabindex="-1"></a><span class="co"># Display the first 5 upregulated genes using kable</span></span>
<span id="cb15-12"><a href="#cb15-12" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(</span>
<span id="cb15-13"><a href="#cb15-13" tabindex="-1"></a>    <span class="fu">head</span>(upregulated_genes),</span>
<span id="cb15-14"><a href="#cb15-14" tabindex="-1"></a>    <span class="at">type =</span> <span class="st">&quot;html&quot;</span>,</span>
<span id="cb15-15"><a href="#cb15-15" tabindex="-1"></a>    <span class="at">row.names =</span> <span class="cn">TRUE</span>,</span>
<span id="cb15-16"><a href="#cb15-16" tabindex="-1"></a>    <span class="at">caption =</span> <span class="st">&quot;Upregulated Genes&quot;</span></span>
<span id="cb15-17"><a href="#cb15-17" tabindex="-1"></a>)</span></code></pre></div>
<table>
<caption>Upregulated Genes</caption>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">logFC</th>
<th align="right">PValue</th>
<th align="right">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">FADD</td>
<td align="right">1.0394283</td>
<td align="right">3.0e-07</td>
<td align="right">0.0029767</td>
</tr>
<tr class="even">
<td align="left">FTH1</td>
<td align="right">1.2325018</td>
<td align="right">3.0e-07</td>
<td align="right">0.0029767</td>
</tr>
<tr class="odd">
<td align="left">CEBPB</td>
<td align="right">0.6365789</td>
<td align="right">7.0e-07</td>
<td align="right">0.0037622</td>
</tr>
<tr class="even">
<td align="left">CEBPD</td>
<td align="right">1.0691417</td>
<td align="right">1.1e-06</td>
<td align="right">0.0037622</td>
</tr>
<tr class="odd">
<td align="left">JUND</td>
<td align="right">0.9522736</td>
<td align="right">1.5e-06</td>
<td align="right">0.0037622</td>
</tr>
<tr class="even">
<td align="left">JUNB</td>
<td align="right">0.9392233</td>
<td align="right">3.2e-06</td>
<td align="right">0.0059327</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a><span class="co"># Get which genes are significantly downregulated</span></span>
<span id="cb16-2"><a href="#cb16-2" tabindex="-1"></a>downregulated_genes <span class="ot">&lt;-</span> <span class="fu">which</span>(top_cbd_infect_vs_cbd_healthy<span class="sc">$</span>table<span class="sc">$</span>logFC <span class="sc">&lt;</span> <span class="dv">0</span> <span class="sc">&amp;</span> top_cbd_infect_vs_cbd_healthy<span class="sc">$</span>table<span class="sc">$</span>FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb16-3"><a href="#cb16-3" tabindex="-1"></a>downregulated_genes <span class="ot">&lt;-</span> top_cbd_infect_vs_cbd_healthy[downregulated_genes,][, <span class="fu">c</span>(<span class="st">&quot;logFC&quot;</span>, <span class="st">&quot;PValue&quot;</span>, <span class="st">&quot;FDR&quot;</span>)]</span>
<span id="cb16-4"><a href="#cb16-4" tabindex="-1"></a></span>
<span id="cb16-5"><a href="#cb16-5" tabindex="-1"></a><span class="co"># Create a data frame of the downregulated genes</span></span>
<span id="cb16-6"><a href="#cb16-6" tabindex="-1"></a>downregulated_genes <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb16-7"><a href="#cb16-7" tabindex="-1"></a>    downregulated_genes,</span>
<span id="cb16-8"><a href="#cb16-8" tabindex="-1"></a>    <span class="at">row.names =</span> <span class="fu">rownames</span>(downregulated_genes)</span>
<span id="cb16-9"><a href="#cb16-9" tabindex="-1"></a>)</span>
<span id="cb16-10"><a href="#cb16-10" tabindex="-1"></a></span>
<span id="cb16-11"><a href="#cb16-11" tabindex="-1"></a><span class="co"># Display the first 5 downregulated genes using kable</span></span>
<span id="cb16-12"><a href="#cb16-12" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(</span>
<span id="cb16-13"><a href="#cb16-13" tabindex="-1"></a>    <span class="fu">head</span>(downregulated_genes),</span>
<span id="cb16-14"><a href="#cb16-14" tabindex="-1"></a>    <span class="at">type =</span> <span class="st">&quot;html&quot;</span>,</span>
<span id="cb16-15"><a href="#cb16-15" tabindex="-1"></a>    <span class="at">row.names =</span> <span class="cn">TRUE</span>,</span>
<span id="cb16-16"><a href="#cb16-16" tabindex="-1"></a>    <span class="at">caption =</span> <span class="st">&quot;Downregulated Genes&quot;</span></span>
<span id="cb16-17"><a href="#cb16-17" tabindex="-1"></a>)</span></code></pre></div>
<table>
<caption>Downregulated Genes</caption>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">logFC</th>
<th align="right">PValue</th>
<th align="right">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">ANXA2</td>
<td align="right">-0.4313385</td>
<td align="right">1.2e-06</td>
<td align="right">0.0037622</td>
</tr>
<tr class="even">
<td align="left">TUBA4A</td>
<td align="right">-0.4940317</td>
<td align="right">1.3e-06</td>
<td align="right">0.0037622</td>
</tr>
<tr class="odd">
<td align="left">POLD2</td>
<td align="right">-0.4632461</td>
<td align="right">1.4e-06</td>
<td align="right">0.0037622</td>
</tr>
<tr class="even">
<td align="left">CAV1</td>
<td align="right">-0.4974154</td>
<td align="right">1.9e-06</td>
<td align="right">0.0043178</td>
</tr>
<tr class="odd">
<td align="left">PSMB6</td>
<td align="right">-0.4145320</td>
<td align="right">2.5e-06</td>
<td align="right">0.0050786</td>
</tr>
<tr class="even">
<td align="left">TUBA1B</td>
<td align="right">-0.4574311</td>
<td align="right">5.3e-06</td>
<td align="right">0.0068803</td>
</tr>
</tbody>
</table>
</div>
<div id="querying-gprofiler" class="section level2 tabset tabset-fade">
<h2 class="tabset tabset-fade">Querying g:Profiler</h2>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" tabindex="-1"></a><span class="co"># Get vectors of the upregulated and downregulated genes</span></span>
<span id="cb17-2"><a href="#cb17-2" tabindex="-1"></a>upreg <span class="ot">&lt;-</span> <span class="fu">rownames</span>(upregulated_genes)</span>
<span id="cb17-3"><a href="#cb17-3" tabindex="-1"></a>downreg <span class="ot">&lt;-</span> <span class="fu">rownames</span>(downregulated_genes)</span>
<span id="cb17-4"><a href="#cb17-4" tabindex="-1"></a></span>
<span id="cb17-5"><a href="#cb17-5" tabindex="-1"></a><span class="co"># Query g:Profiler</span></span>
<span id="cb17-6"><a href="#cb17-6" tabindex="-1"></a>gprofiler_results <span class="ot">&lt;-</span> gprofiler2<span class="sc">::</span><span class="fu">gost</span>(</span>
<span id="cb17-7"><a href="#cb17-7" tabindex="-1"></a>    <span class="at">query =</span> <span class="fu">list</span>(</span>
<span id="cb17-8"><a href="#cb17-8" tabindex="-1"></a>        <span class="at">up =</span> upreg,</span>
<span id="cb17-9"><a href="#cb17-9" tabindex="-1"></a>        <span class="at">down =</span> downreg</span>
<span id="cb17-10"><a href="#cb17-10" tabindex="-1"></a>    ),</span>
<span id="cb17-11"><a href="#cb17-11" tabindex="-1"></a>    <span class="at">organism =</span> <span class="st">&quot;hsapiens&quot;</span>,</span>
<span id="cb17-12"><a href="#cb17-12" tabindex="-1"></a>    <span class="at">sources =</span> <span class="fu">c</span>(<span class="st">&quot;GO:BP&quot;</span>, <span class="st">&quot;REAC&quot;</span>, <span class="st">&quot;WP&quot;</span>),</span>
<span id="cb17-13"><a href="#cb17-13" tabindex="-1"></a>    <span class="at">correction_method =</span> <span class="st">&quot;fdr&quot;</span>,</span>
<span id="cb17-14"><a href="#cb17-14" tabindex="-1"></a>    <span class="at">exclude_iea =</span> <span class="cn">TRUE</span>,</span>
<span id="cb17-15"><a href="#cb17-15" tabindex="-1"></a>    <span class="at">ordered_query =</span> <span class="cn">FALSE</span>,</span>
<span id="cb17-16"><a href="#cb17-16" tabindex="-1"></a>    <span class="at">significant =</span> <span class="cn">FALSE</span></span>
<span id="cb17-17"><a href="#cb17-17" tabindex="-1"></a>)</span>
<span id="cb17-18"><a href="#cb17-18" tabindex="-1"></a></span>
<span id="cb17-19"><a href="#cb17-19" tabindex="-1"></a><span class="co"># Query g:Profiler (upregulated only)</span></span>
<span id="cb17-20"><a href="#cb17-20" tabindex="-1"></a>gprofiler_results_upreg <span class="ot">&lt;-</span> gprofiler2<span class="sc">::</span><span class="fu">gost</span>(</span>
<span id="cb17-21"><a href="#cb17-21" tabindex="-1"></a>    <span class="at">query =</span> <span class="fu">list</span>(</span>
<span id="cb17-22"><a href="#cb17-22" tabindex="-1"></a>        <span class="at">up =</span> upreg</span>
<span id="cb17-23"><a href="#cb17-23" tabindex="-1"></a>    ),</span>
<span id="cb17-24"><a href="#cb17-24" tabindex="-1"></a>    <span class="at">organism =</span> <span class="st">&quot;hsapiens&quot;</span>,</span>
<span id="cb17-25"><a href="#cb17-25" tabindex="-1"></a>    <span class="at">sources =</span> <span class="fu">c</span>(<span class="st">&quot;GO:BP&quot;</span>, <span class="st">&quot;REAC&quot;</span>, <span class="st">&quot;WP&quot;</span>),</span>
<span id="cb17-26"><a href="#cb17-26" tabindex="-1"></a>    <span class="at">correction_method =</span> <span class="st">&quot;fdr&quot;</span>,</span>
<span id="cb17-27"><a href="#cb17-27" tabindex="-1"></a>    <span class="at">exclude_iea =</span> <span class="cn">TRUE</span>,</span>
<span id="cb17-28"><a href="#cb17-28" tabindex="-1"></a>    <span class="at">ordered_query =</span> <span class="cn">FALSE</span>,</span>
<span id="cb17-29"><a href="#cb17-29" tabindex="-1"></a>    <span class="at">significant =</span> <span class="cn">FALSE</span></span>
<span id="cb17-30"><a href="#cb17-30" tabindex="-1"></a>)</span>
<span id="cb17-31"><a href="#cb17-31" tabindex="-1"></a></span>
<span id="cb17-32"><a href="#cb17-32" tabindex="-1"></a><span class="co"># Query g:Profiler (downregulated only)</span></span>
<span id="cb17-33"><a href="#cb17-33" tabindex="-1"></a>gprofiler_results_downreg <span class="ot">&lt;-</span> gprofiler2<span class="sc">::</span><span class="fu">gost</span>(</span>
<span id="cb17-34"><a href="#cb17-34" tabindex="-1"></a>    <span class="at">query =</span> <span class="fu">list</span>(</span>
<span id="cb17-35"><a href="#cb17-35" tabindex="-1"></a>        <span class="at">down =</span> downreg</span>
<span id="cb17-36"><a href="#cb17-36" tabindex="-1"></a>    ),</span>
<span id="cb17-37"><a href="#cb17-37" tabindex="-1"></a>    <span class="at">organism =</span> <span class="st">&quot;hsapiens&quot;</span>,</span>
<span id="cb17-38"><a href="#cb17-38" tabindex="-1"></a>    <span class="at">sources =</span> <span class="fu">c</span>(<span class="st">&quot;GO:BP&quot;</span>, <span class="st">&quot;REAC&quot;</span>, <span class="st">&quot;WP&quot;</span>),</span>
<span id="cb17-39"><a href="#cb17-39" tabindex="-1"></a>    <span class="at">correction_method =</span> <span class="st">&quot;fdr&quot;</span>,</span>
<span id="cb17-40"><a href="#cb17-40" tabindex="-1"></a>    <span class="at">exclude_iea =</span> <span class="cn">TRUE</span>,</span>
<span id="cb17-41"><a href="#cb17-41" tabindex="-1"></a>    <span class="at">ordered_query =</span> <span class="cn">FALSE</span>,</span>
<span id="cb17-42"><a href="#cb17-42" tabindex="-1"></a>    <span class="at">significant =</span> <span class="cn">FALSE</span></span>
<span id="cb17-43"><a href="#cb17-43" tabindex="-1"></a>)</span>
<span id="cb17-44"><a href="#cb17-44" tabindex="-1"></a></span>
<span id="cb17-45"><a href="#cb17-45" tabindex="-1"></a><span class="co"># Rename the results</span></span>
<span id="cb17-46"><a href="#cb17-46" tabindex="-1"></a>results <span class="ot">&lt;-</span> gprofiler_results<span class="sc">$</span>result</span>
<span id="cb17-47"><a href="#cb17-47" tabindex="-1"></a></span>
<span id="cb17-48"><a href="#cb17-48" tabindex="-1"></a><span class="co"># Subset the upregulated results</span></span>
<span id="cb17-49"><a href="#cb17-49" tabindex="-1"></a>upreg_results <span class="ot">&lt;-</span> results[results<span class="sc">$</span>query <span class="sc">==</span> <span class="st">&quot;up&quot;</span>, ]</span>
<span id="cb17-50"><a href="#cb17-50" tabindex="-1"></a>upreg_results_only <span class="ot">&lt;-</span> gprofiler_results_upreg<span class="sc">$</span>result</span>
<span id="cb17-51"><a href="#cb17-51" tabindex="-1"></a></span>
<span id="cb17-52"><a href="#cb17-52" tabindex="-1"></a><span class="co"># Subset the downregulated results</span></span>
<span id="cb17-53"><a href="#cb17-53" tabindex="-1"></a>downreg_results <span class="ot">&lt;-</span> results[results<span class="sc">$</span>query <span class="sc">==</span> <span class="st">&quot;down&quot;</span>, ]</span>
<span id="cb17-54"><a href="#cb17-54" tabindex="-1"></a>downreg_results_only <span class="ot">&lt;-</span> gprofiler_results_downreg<span class="sc">$</span>result</span></code></pre></div>
<div id="overview" class="section level3">
<h3>Overview</h3>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a><span class="co"># Create a data frame to store the number of upregulated and downregulated genes</span></span>
<span id="cb18-2"><a href="#cb18-2" tabindex="-1"></a>gene_counts <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb18-3"><a href="#cb18-3" tabindex="-1"></a>    <span class="at">Upregulated =</span> <span class="fu">nrow</span>(upreg_results),</span>
<span id="cb18-4"><a href="#cb18-4" tabindex="-1"></a>    <span class="st">`</span><span class="at">Upregulated (Only)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">nrow</span>(upreg_results_only),</span>
<span id="cb18-5"><a href="#cb18-5" tabindex="-1"></a>    <span class="at">Downregulated =</span> <span class="fu">nrow</span>(downreg_results),</span>
<span id="cb18-6"><a href="#cb18-6" tabindex="-1"></a>    <span class="st">`</span><span class="at">Downregulated (Only)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">nrow</span>(downreg_results_only)</span>
<span id="cb18-7"><a href="#cb18-7" tabindex="-1"></a>)</span>
<span id="cb18-8"><a href="#cb18-8" tabindex="-1"></a></span>
<span id="cb18-9"><a href="#cb18-9" tabindex="-1"></a><span class="co"># Display the overview of the g:Profiler results using kable</span></span>
<span id="cb18-10"><a href="#cb18-10" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(</span>
<span id="cb18-11"><a href="#cb18-11" tabindex="-1"></a>    gene_counts,</span>
<span id="cb18-12"><a href="#cb18-12" tabindex="-1"></a>    <span class="at">type =</span> <span class="st">&quot;html&quot;</span>,</span>
<span id="cb18-13"><a href="#cb18-13" tabindex="-1"></a>    <span class="at">row.names =</span> <span class="cn">TRUE</span>,</span>
<span id="cb18-14"><a href="#cb18-14" tabindex="-1"></a>    <span class="at">caption =</span> <span class="st">&quot;Overview of g:Profiler Results&quot;</span></span>
<span id="cb18-15"><a href="#cb18-15" tabindex="-1"></a>)</span></code></pre></div>
<table>
<caption>Overview of g:Profiler Results</caption>
<colgroup>
<col width="4%" />
<col width="17%" />
<col width="27%" />
<col width="20%" />
<col width="30%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">Upregulated</th>
<th align="right">Upregulated..Only.</th>
<th align="right">Downregulated</th>
<th align="right">Downregulated..Only.</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">1</td>
<td align="right">8853</td>
<td align="right">8853</td>
<td align="right">6604</td>
<td align="right">6604</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" tabindex="-1"></a><span class="co"># Get the number of terms from each source</span></span>
<span id="cb19-2"><a href="#cb19-2" tabindex="-1"></a>sources <span class="ot">&lt;-</span> <span class="fu">table</span>(results<span class="sc">$</span>source)</span>
<span id="cb19-3"><a href="#cb19-3" tabindex="-1"></a>sources <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(sources)</span>
<span id="cb19-4"><a href="#cb19-4" tabindex="-1"></a><span class="fu">colnames</span>(sources) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Source&quot;</span>, <span class="st">&quot;Number of Terms&quot;</span>)</span>
<span id="cb19-5"><a href="#cb19-5" tabindex="-1"></a></span>
<span id="cb19-6"><a href="#cb19-6" tabindex="-1"></a><span class="co"># Display the number of terms from each source using kable</span></span>
<span id="cb19-7"><a href="#cb19-7" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(</span>
<span id="cb19-8"><a href="#cb19-8" tabindex="-1"></a>    sources,</span>
<span id="cb19-9"><a href="#cb19-9" tabindex="-1"></a>    <span class="at">type =</span> <span class="st">&quot;html&quot;</span>,</span>
<span id="cb19-10"><a href="#cb19-10" tabindex="-1"></a>    <span class="at">row.names =</span> <span class="cn">FALSE</span>,</span>
<span id="cb19-11"><a href="#cb19-11" tabindex="-1"></a>    <span class="at">caption =</span> <span class="st">&quot;Number of Terms from Each Source (Both)&quot;</span></span>
<span id="cb19-12"><a href="#cb19-12" tabindex="-1"></a>)</span></code></pre></div>
<table>
<caption>Number of Terms from Each Source (Both)</caption>
<thead>
<tr class="header">
<th align="left">Source</th>
<th align="right">Number of Terms</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><a href="GO:BP" class="uri">GO:BP</a></td>
<td align="right">11651</td>
</tr>
<tr class="even">
<td align="left">REAC</td>
<td align="right">2688</td>
</tr>
<tr class="odd">
<td align="left">WP</td>
<td align="right">1118</td>
</tr>
</tbody>
</table>
</div>
<div id="upregulated-results" class="section level3">
<h3>Upregulated Results</h3>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" tabindex="-1"></a><span class="co"># Display the upregulated results using kable</span></span>
<span id="cb20-2"><a href="#cb20-2" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(</span>
<span id="cb20-3"><a href="#cb20-3" tabindex="-1"></a>    <span class="fu">head</span>(upreg_results),</span>
<span id="cb20-4"><a href="#cb20-4" tabindex="-1"></a>    <span class="at">type =</span> <span class="st">&quot;html&quot;</span>,</span>
<span id="cb20-5"><a href="#cb20-5" tabindex="-1"></a>    <span class="at">row.names =</span> <span class="cn">FALSE</span>,</span>
<span id="cb20-6"><a href="#cb20-6" tabindex="-1"></a>    <span class="at">caption =</span> <span class="st">&quot;Upregulated Results from g:Profiler&quot;</span></span>
<span id="cb20-7"><a href="#cb20-7" tabindex="-1"></a>)</span></code></pre></div>
<table style="width:100%;">
<caption>Upregulated Results from g:Profiler</caption>
<colgroup>
<col width="2%" />
<col width="5%" />
<col width="3%" />
<col width="4%" />
<col width="5%" />
<col width="8%" />
<col width="4%" />
<col width="4%" />
<col width="5%" />
<col width="3%" />
<col width="29%" />
<col width="10%" />
<col width="6%" />
<col width="6%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">query</th>
<th align="left">significant</th>
<th align="right">p_value</th>
<th align="right">term_size</th>
<th align="right">query_size</th>
<th align="right">intersection_size</th>
<th align="right">precision</th>
<th align="right">recall</th>
<th align="left">term_id</th>
<th align="left">source</th>
<th align="left">term_name</th>
<th align="right">effective_domain_size</th>
<th align="right">source_order</th>
<th align="left">parents</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">up</td>
<td align="left">TRUE</td>
<td align="right">0</td>
<td align="right">3317</td>
<td align="right">1291</td>
<td align="right">540</td>
<td align="right">0.4182804</td>
<td align="right">0.1627977</td>
<td align="left"><a href="GO:0051252" class="uri">GO:0051252</a></td>
<td align="left"><a href="GO:BP" class="uri">GO:BP</a></td>
<td align="left">regulation of RNA metabolic process</td>
<td align="right">16178</td>
<td align="right">14047</td>
<td align="left"><a href="GO:00160" class="uri">GO:00160</a>….</td>
</tr>
<tr class="even">
<td align="left">up</td>
<td align="left">TRUE</td>
<td align="right">0</td>
<td align="right">3592</td>
<td align="right">1291</td>
<td align="right">564</td>
<td align="right">0.4368706</td>
<td align="right">0.1570156</td>
<td align="left"><a href="GO:0019219" class="uri">GO:0019219</a></td>
<td align="left"><a href="GO:BP" class="uri">GO:BP</a></td>
<td align="left">regulation of nucleobase-containing compound metabolic
process</td>
<td align="right">16178</td>
<td align="right">5786</td>
<td align="left"><a href="GO:00061" class="uri">GO:00061</a>….</td>
</tr>
<tr class="odd">
<td align="left">up</td>
<td align="left">TRUE</td>
<td align="right">0</td>
<td align="right">4027</td>
<td align="right">1291</td>
<td align="right">597</td>
<td align="right">0.4624322</td>
<td align="right">0.1482493</td>
<td align="left"><a href="GO:0016070" class="uri">GO:0016070</a></td>
<td align="left"><a href="GO:BP" class="uri">GO:BP</a></td>
<td align="left">RNA metabolic process</td>
<td align="right">16178</td>
<td align="right">5158</td>
<td align="left"><a href="GO:0090304" class="uri">GO:0090304</a></td>
</tr>
<tr class="even">
<td align="left">up</td>
<td align="left">TRUE</td>
<td align="right">0</td>
<td align="right">3173</td>
<td align="right">1291</td>
<td align="right">512</td>
<td align="right">0.3965918</td>
<td align="right">0.1613615</td>
<td align="left"><a href="GO:0032774" class="uri">GO:0032774</a></td>
<td align="left"><a href="GO:BP" class="uri">GO:BP</a></td>
<td align="left">RNA biosynthetic process</td>
<td align="right">16178</td>
<td align="right">8092</td>
<td align="left"><a href="GO:00090" class="uri">GO:00090</a>….</td>
</tr>
<tr class="odd">
<td align="left">up</td>
<td align="left">TRUE</td>
<td align="right">0</td>
<td align="right">3143</td>
<td align="right">1291</td>
<td align="right">509</td>
<td align="right">0.3942680</td>
<td align="right">0.1619472</td>
<td align="left"><a href="GO:0006351" class="uri">GO:0006351</a></td>
<td align="left"><a href="GO:BP" class="uri">GO:BP</a></td>
<td align="left">DNA-templated transcription</td>
<td align="right">16178</td>
<td align="right">2151</td>
<td align="left"><a href="GO:00104" class="uri">GO:00104</a>….</td>
</tr>
<tr class="even">
<td align="left">up</td>
<td align="left">TRUE</td>
<td align="right">0</td>
<td align="right">5297</td>
<td align="right">1291</td>
<td align="right">713</td>
<td align="right">0.5522851</td>
<td align="right">0.1346045</td>
<td align="left"><a href="GO:0060255" class="uri">GO:0060255</a></td>
<td align="left"><a href="GO:BP" class="uri">GO:BP</a></td>
<td align="left">regulation of macromolecule metabolic process</td>
<td align="right">16178</td>
<td align="right">14938</td>
<td align="left"><a href="GO:00192" class="uri">GO:00192</a>….</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a><span class="co"># Get the number of terms from each source</span></span>
<span id="cb21-2"><a href="#cb21-2" tabindex="-1"></a>upreg_sources <span class="ot">&lt;-</span> <span class="fu">table</span>(upreg_results<span class="sc">$</span>source)</span>
<span id="cb21-3"><a href="#cb21-3" tabindex="-1"></a>upreg_sources <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(upreg_sources)</span>
<span id="cb21-4"><a href="#cb21-4" tabindex="-1"></a><span class="fu">colnames</span>(upreg_sources) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Source&quot;</span>, <span class="st">&quot;Number of Terms&quot;</span>)</span>
<span id="cb21-5"><a href="#cb21-5" tabindex="-1"></a></span>
<span id="cb21-6"><a href="#cb21-6" tabindex="-1"></a><span class="co"># Display the number of terms from each source using kable</span></span>
<span id="cb21-7"><a href="#cb21-7" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(</span>
<span id="cb21-8"><a href="#cb21-8" tabindex="-1"></a>    upreg_sources,</span>
<span id="cb21-9"><a href="#cb21-9" tabindex="-1"></a>    <span class="at">type =</span> <span class="st">&quot;html&quot;</span>,</span>
<span id="cb21-10"><a href="#cb21-10" tabindex="-1"></a>    <span class="at">row.names =</span> <span class="cn">FALSE</span>,</span>
<span id="cb21-11"><a href="#cb21-11" tabindex="-1"></a>    <span class="at">caption =</span> <span class="st">&quot;Number of Terms from Each Source (Upregulated)&quot;</span></span>
<span id="cb21-12"><a href="#cb21-12" tabindex="-1"></a>)</span></code></pre></div>
<table>
<caption>Number of Terms from Each Source (Upregulated)</caption>
<thead>
<tr class="header">
<th align="left">Source</th>
<th align="right">Number of Terms</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><a href="GO:BP" class="uri">GO:BP</a></td>
<td align="right">6774</td>
</tr>
<tr class="even">
<td align="left">REAC</td>
<td align="right">1464</td>
</tr>
<tr class="odd">
<td align="left">WP</td>
<td align="right">615</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" tabindex="-1"></a><span class="co"># Get the number of terms from each source</span></span>
<span id="cb22-2"><a href="#cb22-2" tabindex="-1"></a>upreg_sources_only <span class="ot">&lt;-</span> <span class="fu">table</span>(upreg_results_only<span class="sc">$</span>source)</span>
<span id="cb22-3"><a href="#cb22-3" tabindex="-1"></a>upreg_sources_only <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(upreg_sources_only)</span>
<span id="cb22-4"><a href="#cb22-4" tabindex="-1"></a><span class="fu">colnames</span>(upreg_sources_only) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Source&quot;</span>, <span class="st">&quot;Number of Terms&quot;</span>)</span>
<span id="cb22-5"><a href="#cb22-5" tabindex="-1"></a></span>
<span id="cb22-6"><a href="#cb22-6" tabindex="-1"></a><span class="co"># Display the number of terms from each source using kable</span></span>
<span id="cb22-7"><a href="#cb22-7" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(</span>
<span id="cb22-8"><a href="#cb22-8" tabindex="-1"></a>    upreg_sources_only,</span>
<span id="cb22-9"><a href="#cb22-9" tabindex="-1"></a>    <span class="at">type =</span> <span class="st">&quot;html&quot;</span>,</span>
<span id="cb22-10"><a href="#cb22-10" tabindex="-1"></a>    <span class="at">row.names =</span> <span class="cn">FALSE</span>,</span>
<span id="cb22-11"><a href="#cb22-11" tabindex="-1"></a>    <span class="at">caption =</span> <span class="st">&quot;Number of Terms from Each Source (Upregulated Only)&quot;</span></span>
<span id="cb22-12"><a href="#cb22-12" tabindex="-1"></a>)</span></code></pre></div>
<table>
<caption>Number of Terms from Each Source (Upregulated Only)</caption>
<thead>
<tr class="header">
<th align="left">Source</th>
<th align="right">Number of Terms</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><a href="GO:BP" class="uri">GO:BP</a></td>
<td align="right">6774</td>
</tr>
<tr class="even">
<td align="left">REAC</td>
<td align="right">1464</td>
</tr>
<tr class="odd">
<td align="left">WP</td>
<td align="right">615</td>
</tr>
</tbody>
</table>
</div>
<div id="downregulated-results" class="section level3">
<h3>Downregulated Results</h3>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" tabindex="-1"></a><span class="co"># Display the downregulated results using kable</span></span>
<span id="cb23-2"><a href="#cb23-2" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(</span>
<span id="cb23-3"><a href="#cb23-3" tabindex="-1"></a>    <span class="fu">head</span>(downreg_results),</span>
<span id="cb23-4"><a href="#cb23-4" tabindex="-1"></a>    <span class="at">type =</span> <span class="st">&quot;html&quot;</span>,</span>
<span id="cb23-5"><a href="#cb23-5" tabindex="-1"></a>    <span class="at">row.names =</span> <span class="cn">FALSE</span>,</span>
<span id="cb23-6"><a href="#cb23-6" tabindex="-1"></a>    <span class="at">caption =</span> <span class="st">&quot;Downregulated Results from g:Profiler&quot;</span></span>
<span id="cb23-7"><a href="#cb23-7" tabindex="-1"></a>)</span></code></pre></div>
<table>
<caption>Downregulated Results from g:Profiler</caption>
<colgroup>
<col width="3%" />
<col width="6%" />
<col width="4%" />
<col width="5%" />
<col width="5%" />
<col width="9%" />
<col width="5%" />
<col width="5%" />
<col width="5%" />
<col width="3%" />
<col width="19%" />
<col width="11%" />
<col width="6%" />
<col width="6%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">query</th>
<th align="left">significant</th>
<th align="right">p_value</th>
<th align="right">term_size</th>
<th align="right">query_size</th>
<th align="right">intersection_size</th>
<th align="right">precision</th>
<th align="right">recall</th>
<th align="left">term_id</th>
<th align="left">source</th>
<th align="left">term_name</th>
<th align="right">effective_domain_size</th>
<th align="right">source_order</th>
<th align="left">parents</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">down</td>
<td align="left">TRUE</td>
<td align="right">0</td>
<td align="right">790</td>
<td align="right">554</td>
<td align="right">95</td>
<td align="right">0.1714801</td>
<td align="right">0.1202532</td>
<td align="left"><a href="GO:0000278" class="uri">GO:0000278</a></td>
<td align="left"><a href="GO:BP" class="uri">GO:BP</a></td>
<td align="left">mitotic cell cycle</td>
<td align="right">16178</td>
<td align="right">77</td>
<td align="left"><a href="GO:0007049" class="uri">GO:0007049</a></td>
</tr>
<tr class="even">
<td align="left">down</td>
<td align="left">TRUE</td>
<td align="right">0</td>
<td align="right">654</td>
<td align="right">554</td>
<td align="right">81</td>
<td align="right">0.1462094</td>
<td align="right">0.1238532</td>
<td align="left"><a href="GO:1903047" class="uri">GO:1903047</a></td>
<td align="left"><a href="GO:BP" class="uri">GO:BP</a></td>
<td align="left">mitotic cell cycle process</td>
<td align="right">16178</td>
<td align="right">23301</td>
<td align="left"><a href="GO:00002" class="uri">GO:00002</a>….</td>
</tr>
<tr class="odd">
<td align="left">down</td>
<td align="left">TRUE</td>
<td align="right">0</td>
<td align="right">215</td>
<td align="right">554</td>
<td align="right">43</td>
<td align="right">0.0776173</td>
<td align="right">0.2000000</td>
<td align="left"><a href="GO:0000819" class="uri">GO:0000819</a></td>
<td align="left"><a href="GO:BP" class="uri">GO:BP</a></td>
<td align="left">sister chromatid segregation</td>
<td align="right">16178</td>
<td align="right">239</td>
<td align="left"><a href="GO:00512" class="uri">GO:00512</a>….</td>
</tr>
<tr class="even">
<td align="left">down</td>
<td align="left">TRUE</td>
<td align="right">0</td>
<td align="right">176</td>
<td align="right">554</td>
<td align="right">39</td>
<td align="right">0.0703971</td>
<td align="right">0.2215909</td>
<td align="left"><a href="GO:0000070" class="uri">GO:0000070</a></td>
<td align="left"><a href="GO:BP" class="uri">GO:BP</a></td>
<td align="left">mitotic sister chromatid segregation</td>
<td align="right">16178</td>
<td align="right">25</td>
<td align="left"><a href="GO:00008" class="uri">GO:00008</a>….</td>
</tr>
<tr class="odd">
<td align="left">down</td>
<td align="left">TRUE</td>
<td align="right">0</td>
<td align="right">250</td>
<td align="right">554</td>
<td align="right">44</td>
<td align="right">0.0794224</td>
<td align="right">0.1760000</td>
<td align="left"><a href="GO:0140014" class="uri">GO:0140014</a></td>
<td align="left"><a href="GO:BP" class="uri">GO:BP</a></td>
<td align="left">mitotic nuclear division</td>
<td align="right">16178</td>
<td align="right">20198</td>
<td align="left"><a href="GO:00002" class="uri">GO:00002</a>….</td>
</tr>
<tr class="even">
<td align="left">down</td>
<td align="left">TRUE</td>
<td align="right">0</td>
<td align="right">275</td>
<td align="right">554</td>
<td align="right">46</td>
<td align="right">0.0830325</td>
<td align="right">0.1672727</td>
<td align="left"><a href="GO:0098813" class="uri">GO:0098813</a></td>
<td align="left"><a href="GO:BP" class="uri">GO:BP</a></td>
<td align="left">nuclear chromosome segregation</td>
<td align="right">16178</td>
<td align="right">19446</td>
<td align="left"><a href="GO:0007059" class="uri">GO:0007059</a></td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" tabindex="-1"></a><span class="co"># Get the number of terms from each source</span></span>
<span id="cb24-2"><a href="#cb24-2" tabindex="-1"></a>downreg_sources <span class="ot">&lt;-</span> <span class="fu">table</span>(downreg_results<span class="sc">$</span>source)</span>
<span id="cb24-3"><a href="#cb24-3" tabindex="-1"></a>downreg_sources <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(downreg_sources)</span>
<span id="cb24-4"><a href="#cb24-4" tabindex="-1"></a><span class="fu">colnames</span>(downreg_sources) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Source&quot;</span>, <span class="st">&quot;Number of Terms&quot;</span>)</span>
<span id="cb24-5"><a href="#cb24-5" tabindex="-1"></a></span>
<span id="cb24-6"><a href="#cb24-6" tabindex="-1"></a><span class="co"># Display the number of terms from each source using kable</span></span>
<span id="cb24-7"><a href="#cb24-7" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(</span>
<span id="cb24-8"><a href="#cb24-8" tabindex="-1"></a>    downreg_sources,</span>
<span id="cb24-9"><a href="#cb24-9" tabindex="-1"></a>    <span class="at">type =</span> <span class="st">&quot;html&quot;</span>,</span>
<span id="cb24-10"><a href="#cb24-10" tabindex="-1"></a>    <span class="at">row.names =</span> <span class="cn">FALSE</span>,</span>
<span id="cb24-11"><a href="#cb24-11" tabindex="-1"></a>    <span class="at">caption =</span> <span class="st">&quot;Number of Terms from Each Source (Downregulated)&quot;</span></span>
<span id="cb24-12"><a href="#cb24-12" tabindex="-1"></a>)</span></code></pre></div>
<table>
<caption>Number of Terms from Each Source (Downregulated)</caption>
<thead>
<tr class="header">
<th align="left">Source</th>
<th align="right">Number of Terms</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><a href="GO:BP" class="uri">GO:BP</a></td>
<td align="right">4877</td>
</tr>
<tr class="even">
<td align="left">REAC</td>
<td align="right">1224</td>
</tr>
<tr class="odd">
<td align="left">WP</td>
<td align="right">503</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" tabindex="-1"></a><span class="co"># Get the number of terms from each source</span></span>
<span id="cb25-2"><a href="#cb25-2" tabindex="-1"></a>downreg_sources_only <span class="ot">&lt;-</span> <span class="fu">table</span>(downreg_results_only<span class="sc">$</span>source)</span>
<span id="cb25-3"><a href="#cb25-3" tabindex="-1"></a>downreg_sources_only <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(downreg_sources_only)</span>
<span id="cb25-4"><a href="#cb25-4" tabindex="-1"></a><span class="fu">colnames</span>(downreg_sources_only) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Source&quot;</span>, <span class="st">&quot;Number of Terms&quot;</span>)</span>
<span id="cb25-5"><a href="#cb25-5" tabindex="-1"></a></span>
<span id="cb25-6"><a href="#cb25-6" tabindex="-1"></a><span class="co"># Display the number of terms from each source using kable</span></span>
<span id="cb25-7"><a href="#cb25-7" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(</span>
<span id="cb25-8"><a href="#cb25-8" tabindex="-1"></a>    downreg_sources_only,</span>
<span id="cb25-9"><a href="#cb25-9" tabindex="-1"></a>    <span class="at">type =</span> <span class="st">&quot;html&quot;</span>,</span>
<span id="cb25-10"><a href="#cb25-10" tabindex="-1"></a>    <span class="at">row.names =</span> <span class="cn">FALSE</span>,</span>
<span id="cb25-11"><a href="#cb25-11" tabindex="-1"></a>    <span class="at">caption =</span> <span class="st">&quot;Number of Terms from Each Source (Downregulated Only)&quot;</span></span>
<span id="cb25-12"><a href="#cb25-12" tabindex="-1"></a>)</span></code></pre></div>
<table>
<caption>Number of Terms from Each Source (Downregulated Only)</caption>
<thead>
<tr class="header">
<th align="left">Source</th>
<th align="right">Number of Terms</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><a href="GO:BP" class="uri">GO:BP</a></td>
<td align="right">4877</td>
</tr>
<tr class="even">
<td align="left">REAC</td>
<td align="right">1224</td>
</tr>
<tr class="odd">
<td align="left">WP</td>
<td align="right">503</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="over-representation-analysis" class="section level2">
<h2>Over-representation Analysis</h2>
<div id="filtering-the-results"
class="section level3 tabset tabset-fade">
<h3 class="tabset tabset-fade">Filtering the Results</h3>
<p>Since there are 8853 significant terms for the upregulated genes and
6604 significant terms for the downregulated genes, we need to filter
our results to ensure we are only examining relevant terms.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" tabindex="-1"></a><span class="co"># Filter the results to only include results with at least 200 terms</span></span>
<span id="cb26-2"><a href="#cb26-2" tabindex="-1"></a>results_filtered <span class="ot">&lt;-</span> results[results<span class="sc">$</span>term_size <span class="sc">&lt;</span> <span class="dv">500</span>, ]</span>
<span id="cb26-3"><a href="#cb26-3" tabindex="-1"></a>upreg_results_filtered <span class="ot">&lt;-</span> upreg_results[upreg_results<span class="sc">$</span>term_size <span class="sc">&lt;</span> <span class="dv">500</span>, ]</span>
<span id="cb26-4"><a href="#cb26-4" tabindex="-1"></a>upreg_results_only_filtered <span class="ot">&lt;-</span> upreg_results_only[upreg_results_only<span class="sc">$</span>term_size <span class="sc">&lt;</span> <span class="dv">500</span>, ]</span>
<span id="cb26-5"><a href="#cb26-5" tabindex="-1"></a>downreg_results_filtered <span class="ot">&lt;-</span> downreg_results[downreg_results<span class="sc">$</span>term_size <span class="sc">&lt;</span> <span class="dv">500</span>, ]</span>
<span id="cb26-6"><a href="#cb26-6" tabindex="-1"></a>downreg_results_only_filtered <span class="ot">&lt;-</span> downreg_results_only[downreg_results_only<span class="sc">$</span>term_size <span class="sc">&lt;</span> <span class="dv">500</span>, ]</span>
<span id="cb26-7"><a href="#cb26-7" tabindex="-1"></a></span>
<span id="cb26-8"><a href="#cb26-8" tabindex="-1"></a><span class="co"># Display the number of significant terms before and after filtering using kable</span></span>
<span id="cb26-9"><a href="#cb26-9" tabindex="-1"></a>gene_counts_both <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb26-10"><a href="#cb26-10" tabindex="-1"></a>    <span class="at">Regulated =</span> <span class="fu">nrow</span>(results),</span>
<span id="cb26-11"><a href="#cb26-11" tabindex="-1"></a>    <span class="at">Upregulated =</span> <span class="fu">nrow</span>(upreg_results),</span>
<span id="cb26-12"><a href="#cb26-12" tabindex="-1"></a>    <span class="at">Downregulated =</span> <span class="fu">nrow</span>(downreg_results),</span>
<span id="cb26-13"><a href="#cb26-13" tabindex="-1"></a>    <span class="st">`</span><span class="at">Regulated (Filtered)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">nrow</span>(results_filtered),</span>
<span id="cb26-14"><a href="#cb26-14" tabindex="-1"></a>    <span class="st">`</span><span class="at">Upregulated (Filtered)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">nrow</span>(upreg_results_filtered),</span>
<span id="cb26-15"><a href="#cb26-15" tabindex="-1"></a>    <span class="st">`</span><span class="at">Downregulated (Filtered)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">nrow</span>(downreg_results_filtered)</span>
<span id="cb26-16"><a href="#cb26-16" tabindex="-1"></a>)</span>
<span id="cb26-17"><a href="#cb26-17" tabindex="-1"></a><span class="co"># Add column names</span></span>
<span id="cb26-18"><a href="#cb26-18" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(</span>
<span id="cb26-19"><a href="#cb26-19" tabindex="-1"></a>    gene_counts_both,</span>
<span id="cb26-20"><a href="#cb26-20" tabindex="-1"></a>    <span class="at">type =</span> <span class="st">&quot;html&quot;</span>,</span>
<span id="cb26-21"><a href="#cb26-21" tabindex="-1"></a>    <span class="at">row.names =</span> <span class="cn">FALSE</span>,</span>
<span id="cb26-22"><a href="#cb26-22" tabindex="-1"></a>    <span class="at">caption =</span> <span class="st">&quot;Overview of g:Profiler Results&quot;</span></span>
<span id="cb26-23"><a href="#cb26-23" tabindex="-1"></a>)</span></code></pre></div>
<table>
<caption>Overview of g:Profiler Results</caption>
<colgroup>
<col width="9%" />
<col width="11%" />
<col width="13%" />
<col width="20%" />
<col width="21%" />
<col width="23%" />
</colgroup>
<thead>
<tr class="header">
<th align="right">Regulated</th>
<th align="right">Upregulated</th>
<th align="right">Downregulated</th>
<th align="right">Regulated..Filtered.</th>
<th align="right">Upregulated..Filtered.</th>
<th align="right">Downregulated..Filtered.</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">15457</td>
<td align="right">8853</td>
<td align="right">6604</td>
<td align="right">14659</td>
<td align="right">8454</td>
<td align="right">6205</td>
</tr>
</tbody>
</table>
<p>We see that filtering the results has reduced the number of
significant terms by 399 for the upregulated genes and 399 for the
downregulated genes.</p>
<div id="overview-filtered" class="section level4">
<h4>Overview (Filtered)</h4>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" tabindex="-1"></a><span class="co"># Get the column names we want to display</span></span>
<span id="cb27-2"><a href="#cb27-2" tabindex="-1"></a>reg_cols <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;term_id&quot;</span>, <span class="st">&quot;term_name&quot;</span>, <span class="st">&quot;p_value&quot;</span>, <span class="st">&quot;intersection_size&quot;</span>, <span class="st">&quot;term_size&quot;</span>)</span>
<span id="cb27-3"><a href="#cb27-3" tabindex="-1"></a></span>
<span id="cb27-4"><a href="#cb27-4" tabindex="-1"></a><span class="co"># Display the filtered overview of the g:Profiler results using kable</span></span>
<span id="cb27-5"><a href="#cb27-5" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(</span>
<span id="cb27-6"><a href="#cb27-6" tabindex="-1"></a>    <span class="fu">head</span>(results_filtered[, reg_cols], <span class="dv">10</span>),</span>
<span id="cb27-7"><a href="#cb27-7" tabindex="-1"></a>    <span class="at">type =</span> <span class="st">&quot;html&quot;</span>,</span>
<span id="cb27-8"><a href="#cb27-8" tabindex="-1"></a>    <span class="at">row.names =</span> <span class="cn">FALSE</span>,</span>
<span id="cb27-9"><a href="#cb27-9" tabindex="-1"></a>    <span class="at">caption =</span> <span class="st">&quot;Filtered Overview of g:Profiler Results&quot;</span></span>
<span id="cb27-10"><a href="#cb27-10" tabindex="-1"></a>)</span></code></pre></div>
<table>
<caption>Filtered Overview of g:Profiler Results</caption>
<colgroup>
<col width="11%" />
<col width="51%" />
<col width="8%" />
<col width="18%" />
<col width="10%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">term_id</th>
<th align="left">term_name</th>
<th align="right">p_value</th>
<th align="right">intersection_size</th>
<th align="right">term_size</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><a href="GO:0000819" class="uri">GO:0000819</a></td>
<td align="left">sister chromatid segregation</td>
<td align="right">0</td>
<td align="right">43</td>
<td align="right">215</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0000070" class="uri">GO:0000070</a></td>
<td align="left">mitotic sister chromatid segregation</td>
<td align="right">0</td>
<td align="right">39</td>
<td align="right">176</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0140014" class="uri">GO:0140014</a></td>
<td align="left">mitotic nuclear division</td>
<td align="right">0</td>
<td align="right">44</td>
<td align="right">250</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0098813" class="uri">GO:0098813</a></td>
<td align="left">nuclear chromosome segregation</td>
<td align="right">0</td>
<td align="right">46</td>
<td align="right">275</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:1905818" class="uri">GO:1905818</a></td>
<td align="left">regulation of chromosome separation</td>
<td align="right">0</td>
<td align="right">24</td>
<td align="right">70</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0051304" class="uri">GO:0051304</a></td>
<td align="left">chromosome separation</td>
<td align="right">0</td>
<td align="right">24</td>
<td align="right">76</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0007059" class="uri">GO:0007059</a></td>
<td align="left">chromosome segregation</td>
<td align="right">0</td>
<td align="right">49</td>
<td align="right">360</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0051983" class="uri">GO:0051983</a></td>
<td align="left">regulation of chromosome segregation</td>
<td align="right">0</td>
<td align="right">29</td>
<td align="right">124</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0000280" class="uri">GO:0000280</a></td>
<td align="left">nuclear division</td>
<td align="right">0</td>
<td align="right">49</td>
<td align="right">377</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0010965" class="uri">GO:0010965</a></td>
<td align="left">regulation of mitotic sister chromatid separation</td>
<td align="right">0</td>
<td align="right">20</td>
<td align="right">56</td>
</tr>
</tbody>
</table>
</div>
<div id="upregulated-results-filtered" class="section level4">
<h4>Upregulated Results (Filtered)</h4>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" tabindex="-1"></a><span class="co"># Get the column names we want to display</span></span>
<span id="cb28-2"><a href="#cb28-2" tabindex="-1"></a>upreg_cols <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;term_id&quot;</span>, <span class="st">&quot;term_name&quot;</span>, <span class="st">&quot;p_value&quot;</span>, <span class="st">&quot;intersection_size&quot;</span>, <span class="st">&quot;term_size&quot;</span>)</span>
<span id="cb28-3"><a href="#cb28-3" tabindex="-1"></a></span>
<span id="cb28-4"><a href="#cb28-4" tabindex="-1"></a><span class="co"># Sort the results by p-value in ascending order</span></span>
<span id="cb28-5"><a href="#cb28-5" tabindex="-1"></a>upreg_results_filtered <span class="ot">&lt;-</span> upreg_results_filtered[<span class="fu">order</span>(upreg_results_filtered<span class="sc">$</span>p_value), ]</span>
<span id="cb28-6"><a href="#cb28-6" tabindex="-1"></a></span>
<span id="cb28-7"><a href="#cb28-7" tabindex="-1"></a><span class="co"># Display the filtered upregulated results using kable</span></span>
<span id="cb28-8"><a href="#cb28-8" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(</span>
<span id="cb28-9"><a href="#cb28-9" tabindex="-1"></a>    <span class="fu">head</span>(upreg_results_filtered[, upreg_cols], <span class="dv">10</span>),</span>
<span id="cb28-10"><a href="#cb28-10" tabindex="-1"></a>    <span class="at">type =</span> <span class="st">&quot;html&quot;</span>,</span>
<span id="cb28-11"><a href="#cb28-11" tabindex="-1"></a>    <span class="at">row.names =</span> <span class="cn">FALSE</span>,</span>
<span id="cb28-12"><a href="#cb28-12" tabindex="-1"></a>    <span class="at">caption =</span> <span class="st">&quot;Filtered Upregulated Results from g:Profiler&quot;</span></span>
<span id="cb28-13"><a href="#cb28-13" tabindex="-1"></a>)</span></code></pre></div>
<table>
<caption>Filtered Upregulated Results from g:Profiler</caption>
<colgroup>
<col width="8%" />
<col width="61%" />
<col width="6%" />
<col width="14%" />
<col width="8%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">term_id</th>
<th align="left">term_name</th>
<th align="right">p_value</th>
<th align="right">intersection_size</th>
<th align="right">term_size</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><a href="GO:0045596" class="uri">GO:0045596</a></td>
<td align="left">negative regulation of cell differentiation</td>
<td align="right">0e+00</td>
<td align="right">78</td>
<td align="right">434</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0071559" class="uri">GO:0071559</a></td>
<td align="left">response to transforming growth factor beta</td>
<td align="right">0e+00</td>
<td align="right">48</td>
<td align="right">207</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0071560" class="uri">GO:0071560</a></td>
<td align="left">cellular response to transforming growth factor beta
stimulus</td>
<td align="right">0e+00</td>
<td align="right">46</td>
<td align="right">201</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0010586" class="uri">GO:0010586</a></td>
<td align="left">miRNA metabolic process</td>
<td align="right">0e+00</td>
<td align="right">29</td>
<td align="right">92</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0007179" class="uri">GO:0007179</a></td>
<td align="left">transforming growth factor beta receptor signaling
pathway</td>
<td align="right">0e+00</td>
<td align="right">42</td>
<td align="right">175</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:2000628" class="uri">GO:2000628</a></td>
<td align="left">regulation of miRNA metabolic process</td>
<td align="right">0e+00</td>
<td align="right">26</td>
<td align="right">77</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0009896" class="uri">GO:0009896</a></td>
<td align="left">positive regulation of catabolic process</td>
<td align="right">0e+00</td>
<td align="right">78</td>
<td align="right">474</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0017015" class="uri">GO:0017015</a></td>
<td align="left">regulation of transforming growth factor beta receptor
signaling pathway</td>
<td align="right">1e-07</td>
<td align="right">32</td>
<td align="right">122</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0043161" class="uri">GO:0043161</a></td>
<td align="left">proteasome-mediated ubiquitin-dependent protein
catabolic process</td>
<td align="right">1e-07</td>
<td align="right">64</td>
<td align="right">363</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:1903844" class="uri">GO:1903844</a></td>
<td align="left">regulation of cellular response to transforming growth
factor beta stimulus</td>
<td align="right">1e-07</td>
<td align="right">32</td>
<td align="right">124</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" tabindex="-1"></a><span class="co"># Get the number of terms from each source</span></span>
<span id="cb29-2"><a href="#cb29-2" tabindex="-1"></a>upreg_sources <span class="ot">&lt;-</span> <span class="fu">table</span>(upreg_results_filtered<span class="sc">$</span>source)</span>
<span id="cb29-3"><a href="#cb29-3" tabindex="-1"></a>upreg_sources <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(upreg_sources)</span>
<span id="cb29-4"><a href="#cb29-4" tabindex="-1"></a><span class="fu">colnames</span>(upreg_sources) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Source&quot;</span>, <span class="st">&quot;Number of Terms&quot;</span>)</span>
<span id="cb29-5"><a href="#cb29-5" tabindex="-1"></a></span>
<span id="cb29-6"><a href="#cb29-6" tabindex="-1"></a><span class="co"># Display the number of terms from each source using kable</span></span>
<span id="cb29-7"><a href="#cb29-7" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(</span>
<span id="cb29-8"><a href="#cb29-8" tabindex="-1"></a>    upreg_sources,</span>
<span id="cb29-9"><a href="#cb29-9" tabindex="-1"></a>    <span class="at">type =</span> <span class="st">&quot;html&quot;</span>,</span>
<span id="cb29-10"><a href="#cb29-10" tabindex="-1"></a>    <span class="at">row.names =</span> <span class="cn">FALSE</span>,</span>
<span id="cb29-11"><a href="#cb29-11" tabindex="-1"></a>    <span class="at">caption =</span> <span class="st">&quot;Number of Terms from Each Source (Upregulated)&quot;</span></span>
<span id="cb29-12"><a href="#cb29-12" tabindex="-1"></a>)</span></code></pre></div>
<table>
<caption>Number of Terms from Each Source (Upregulated)</caption>
<thead>
<tr class="header">
<th align="left">Source</th>
<th align="right">Number of Terms</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><a href="GO:BP" class="uri">GO:BP</a></td>
<td align="right">6410</td>
</tr>
<tr class="even">
<td align="left">REAC</td>
<td align="right">1430</td>
</tr>
<tr class="odd">
<td align="left">WP</td>
<td align="right">614</td>
</tr>
</tbody>
</table>
</div>
<div id="downregulated-results-filtered" class="section level4">
<h4>Downregulated Results (Filtered)</h4>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" tabindex="-1"></a><span class="co"># Get the column names we want to display</span></span>
<span id="cb30-2"><a href="#cb30-2" tabindex="-1"></a>downreg_cols <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;term_id&quot;</span>, <span class="st">&quot;term_name&quot;</span>, <span class="st">&quot;p_value&quot;</span>, <span class="st">&quot;intersection_size&quot;</span>, <span class="st">&quot;term_size&quot;</span>)</span>
<span id="cb30-3"><a href="#cb30-3" tabindex="-1"></a></span>
<span id="cb30-4"><a href="#cb30-4" tabindex="-1"></a><span class="co"># Sort the results by p-value in ascending order</span></span>
<span id="cb30-5"><a href="#cb30-5" tabindex="-1"></a>downreg_results_filtered <span class="ot">&lt;-</span> downreg_results_filtered[<span class="fu">order</span>(downreg_results_filtered<span class="sc">$</span>p_value), ]</span>
<span id="cb30-6"><a href="#cb30-6" tabindex="-1"></a></span>
<span id="cb30-7"><a href="#cb30-7" tabindex="-1"></a><span class="co"># Display the filtered downregulated results using kable</span></span>
<span id="cb30-8"><a href="#cb30-8" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(</span>
<span id="cb30-9"><a href="#cb30-9" tabindex="-1"></a>    <span class="fu">head</span>(downreg_results_filtered[, downreg_cols], <span class="dv">10</span>),</span>
<span id="cb30-10"><a href="#cb30-10" tabindex="-1"></a>    <span class="at">type =</span> <span class="st">&quot;html&quot;</span>,</span>
<span id="cb30-11"><a href="#cb30-11" tabindex="-1"></a>    <span class="at">row.names =</span> <span class="cn">FALSE</span>,</span>
<span id="cb30-12"><a href="#cb30-12" tabindex="-1"></a>    <span class="at">caption =</span> <span class="st">&quot;Filtered Downregulated Results from g:Profiler&quot;</span></span>
<span id="cb30-13"><a href="#cb30-13" tabindex="-1"></a>)</span></code></pre></div>
<table>
<caption>Filtered Downregulated Results from g:Profiler</caption>
<colgroup>
<col width="20%" />
<col width="40%" />
<col width="8%" />
<col width="19%" />
<col width="10%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">term_id</th>
<th align="left">term_name</th>
<th align="right">p_value</th>
<th align="right">intersection_size</th>
<th align="right">term_size</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><a href="GO:0000819" class="uri">GO:0000819</a></td>
<td align="left">sister chromatid segregation</td>
<td align="right">0</td>
<td align="right">43</td>
<td align="right">215</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0000070" class="uri">GO:0000070</a></td>
<td align="left">mitotic sister chromatid segregation</td>
<td align="right">0</td>
<td align="right">39</td>
<td align="right">176</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0140014" class="uri">GO:0140014</a></td>
<td align="left">mitotic nuclear division</td>
<td align="right">0</td>
<td align="right">44</td>
<td align="right">250</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0098813" class="uri">GO:0098813</a></td>
<td align="left">nuclear chromosome segregation</td>
<td align="right">0</td>
<td align="right">46</td>
<td align="right">275</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:1905818" class="uri">GO:1905818</a></td>
<td align="left">regulation of chromosome separation</td>
<td align="right">0</td>
<td align="right">24</td>
<td align="right">70</td>
</tr>
<tr class="even">
<td align="left">REAC:R-HSA-68882</td>
<td align="left">Mitotic Anaphase</td>
<td align="right">0</td>
<td align="right">43</td>
<td align="right">228</td>
</tr>
<tr class="odd">
<td align="left">REAC:R-HSA-2555396</td>
<td align="left">Mitotic Metaphase and Anaphase</td>
<td align="right">0</td>
<td align="right">43</td>
<td align="right">229</td>
</tr>
<tr class="even">
<td align="left">REAC:R-HSA-2467813</td>
<td align="left">Separation of Sister Chromatids</td>
<td align="right">0</td>
<td align="right">39</td>
<td align="right">188</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0051304" class="uri">GO:0051304</a></td>
<td align="left">chromosome separation</td>
<td align="right">0</td>
<td align="right">24</td>
<td align="right">76</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0007059" class="uri">GO:0007059</a></td>
<td align="left">chromosome segregation</td>
<td align="right">0</td>
<td align="right">49</td>
<td align="right">360</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" tabindex="-1"></a><span class="co"># Get the number of terms from each source</span></span>
<span id="cb31-2"><a href="#cb31-2" tabindex="-1"></a>downreg_sources <span class="ot">&lt;-</span> <span class="fu">table</span>(downreg_results_filtered<span class="sc">$</span>source)</span>
<span id="cb31-3"><a href="#cb31-3" tabindex="-1"></a>downreg_sources <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(downreg_sources)</span>
<span id="cb31-4"><a href="#cb31-4" tabindex="-1"></a><span class="fu">colnames</span>(downreg_sources) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Source&quot;</span>, <span class="st">&quot;Number of Terms&quot;</span>)</span>
<span id="cb31-5"><a href="#cb31-5" tabindex="-1"></a></span>
<span id="cb31-6"><a href="#cb31-6" tabindex="-1"></a><span class="co"># Display the number of terms from each source using kable</span></span>
<span id="cb31-7"><a href="#cb31-7" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(</span>
<span id="cb31-8"><a href="#cb31-8" tabindex="-1"></a>    downreg_sources,</span>
<span id="cb31-9"><a href="#cb31-9" tabindex="-1"></a>    <span class="at">type =</span> <span class="st">&quot;html&quot;</span>,</span>
<span id="cb31-10"><a href="#cb31-10" tabindex="-1"></a>    <span class="at">row.names =</span> <span class="cn">FALSE</span>,</span>
<span id="cb31-11"><a href="#cb31-11" tabindex="-1"></a>    <span class="at">caption =</span> <span class="st">&quot;Number of Terms from Each Source (Downregulated)&quot;</span></span>
<span id="cb31-12"><a href="#cb31-12" tabindex="-1"></a>)</span></code></pre></div>
<table>
<caption>Number of Terms from Each Source (Downregulated)</caption>
<thead>
<tr class="header">
<th align="left">Source</th>
<th align="right">Number of Terms</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><a href="GO:BP" class="uri">GO:BP</a></td>
<td align="right">4513</td>
</tr>
<tr class="even">
<td align="left">REAC</td>
<td align="right">1190</td>
</tr>
<tr class="odd">
<td align="left">WP</td>
<td align="right">502</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="questions-1" class="section level3">
<h3>Questions</h3>
<p><strong>1. Which method did you choose and why?</strong></p>
<p>To perform the over-representation analysis (ORA), I chose to use
g:Profiler. This is a tool that I’d been exposed to earlier in the term
through one of the <a
href="https://github.com/bcb420-2024/Ashenafee_Mandefro/wiki/8-%E2%80%90-Getting-acquainted-with-using-G%3AProfiler">journal
assignments</a>. Although I had experience using it via the web
interface, picking it up for use in this R pipeline wasn’t too
difficult.</p>
<p>From the experience using it before (although briefly), I knew that
this tool offered a gateway to comprehensive functional annotations on
many genes, given its connection to a wide range of databases, such as
<a href="https://www.wikipathways.org/">WikiPathways</a> and <a
href="https://reactome.org/">Reactome</a>, as well as the integration of
field-standards such as Gene Ontology (GO) terms.</p>
<p><strong>2. What annotation data did you use and why? What version of
the annotation are you using?</strong></p>
<p>I used the following annotation sources:</p>
<ul>
<li>Gene Ontology Biological Process (<a href="GO:BP"
class="uri">GO:BP</a>), Version <code>releases/2024-01-17</code></li>
<li>Reactome (REAC), Version <code>2024-1-25</code></li>
<li>WikiPathways (WP), Version <code>20240101</code></li>
</ul>
<p>The <a href="GO:BP" class="uri">GO:BP</a> terms are a standard for
annotating genes with their biological process, regardless of the source
of data. On the other hand, Reactome and WikiPathways are databases that
provide curated pathways for human biology. Furthermore, we were exposed
to these annotation sources via the g:Profiler journal assignment.</p>
<p>The version numbers were found by accessing the <a
href="https://biit.cs.ut.ee/gprofiler/gost">g:Profiler web interface</a>
and checking the version numbers of the annotation sources.</p>
<p><strong>3. How many genesets were returned with what
thresholds?</strong></p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" tabindex="-1"></a>genesets <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb32-2"><a href="#cb32-2" tabindex="-1"></a>    <span class="at">Upregulated =</span> <span class="fu">nrow</span>(upreg_results_filtered),</span>
<span id="cb32-3"><a href="#cb32-3" tabindex="-1"></a>    <span class="at">Downregulated =</span> <span class="fu">nrow</span>(downreg_results_filtered),</span>
<span id="cb32-4"><a href="#cb32-4" tabindex="-1"></a>    <span class="at">Threshold =</span> <span class="dv">500</span></span>
<span id="cb32-5"><a href="#cb32-5" tabindex="-1"></a>)</span>
<span id="cb32-6"><a href="#cb32-6" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(</span>
<span id="cb32-7"><a href="#cb32-7" tabindex="-1"></a>    genesets,</span>
<span id="cb32-8"><a href="#cb32-8" tabindex="-1"></a>    <span class="at">type =</span> <span class="st">&quot;html&quot;</span>,</span>
<span id="cb32-9"><a href="#cb32-9" tabindex="-1"></a>    <span class="at">row.names =</span> <span class="cn">FALSE</span>,</span>
<span id="cb32-10"><a href="#cb32-10" tabindex="-1"></a>    <span class="at">caption =</span> <span class="st">&quot;Number of Genesets Returned with a Threshold of 500&quot;</span></span>
<span id="cb32-11"><a href="#cb32-11" tabindex="-1"></a>)</span></code></pre></div>
<table>
<caption>Number of Genesets Returned with a Threshold of 500</caption>
<thead>
<tr class="header">
<th align="right">Upregulated</th>
<th align="right">Downregulated</th>
<th align="right">Threshold</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">8454</td>
<td align="right">6205</td>
<td align="right">500</td>
</tr>
</tbody>
</table>
<p><strong>4. How do results differ if you were to run the analysis on
the entire dataset instead of on upregulated and downregulated genes
separately?</strong></p>
<!-- link to section ### Upregulated Results -->
<p>A run of g:Profiler using the entire dataset would return 8853
significant terms for the upregulated genes and 6604 significant terms
for the downregulated genes. However, querying for upregulated genes
only would return 8853 significant terms, and querying for downregulated
genes only would return 6604 significant terms.</p>
<p>There is no difference in the total number of terms returned, with
the counts of each annotation source being retained regardless of the
method used as well:</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" tabindex="-1"></a><span class="co"># Get the sources for upreg_results_only_filtered</span></span>
<span id="cb33-2"><a href="#cb33-2" tabindex="-1"></a>upreg_sources_only <span class="ot">&lt;-</span> <span class="fu">table</span>(upreg_results_only_filtered<span class="sc">$</span>source)</span>
<span id="cb33-3"><a href="#cb33-3" tabindex="-1"></a>upreg_sources_only <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(upreg_sources_only)</span>
<span id="cb33-4"><a href="#cb33-4" tabindex="-1"></a><span class="fu">colnames</span>(upreg_sources_only) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Source&quot;</span>, <span class="st">&quot;Number of Terms&quot;</span>)</span>
<span id="cb33-5"><a href="#cb33-5" tabindex="-1"></a></span>
<span id="cb33-6"><a href="#cb33-6" tabindex="-1"></a><span class="co"># Get the sources for downreg_results_only_filtered</span></span>
<span id="cb33-7"><a href="#cb33-7" tabindex="-1"></a>downreg_sources_only <span class="ot">&lt;-</span> <span class="fu">table</span>(downreg_results_only_filtered<span class="sc">$</span>source)</span>
<span id="cb33-8"><a href="#cb33-8" tabindex="-1"></a>downreg_sources_only <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(downreg_sources_only)</span>
<span id="cb33-9"><a href="#cb33-9" tabindex="-1"></a><span class="fu">colnames</span>(downreg_sources_only) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Source&quot;</span>, <span class="st">&quot;Number of Terms&quot;</span>)</span>
<span id="cb33-10"><a href="#cb33-10" tabindex="-1"></a></span>
<span id="cb33-11"><a href="#cb33-11" tabindex="-1"></a><span class="co"># Create a data frame to store the number of terms from each source</span></span>
<span id="cb33-12"><a href="#cb33-12" tabindex="-1"></a>sources_comparison <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb33-13"><a href="#cb33-13" tabindex="-1"></a>    <span class="at">Source =</span> <span class="fu">c</span>(<span class="st">&quot;GO:BP&quot;</span>, <span class="st">&quot;REAC&quot;</span>, <span class="st">&quot;WP&quot;</span>),</span>
<span id="cb33-14"><a href="#cb33-14" tabindex="-1"></a>    <span class="st">`</span><span class="at">NoT (Upregulated)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(</span>
<span id="cb33-15"><a href="#cb33-15" tabindex="-1"></a>        upreg_sources[upreg_sources<span class="sc">$</span>Source <span class="sc">==</span> <span class="st">&quot;GO:BP&quot;</span>, <span class="st">&quot;Number of Terms&quot;</span>],</span>
<span id="cb33-16"><a href="#cb33-16" tabindex="-1"></a>        upreg_sources[upreg_sources<span class="sc">$</span>Source <span class="sc">==</span> <span class="st">&quot;REAC&quot;</span>, <span class="st">&quot;Number of Terms&quot;</span>],</span>
<span id="cb33-17"><a href="#cb33-17" tabindex="-1"></a>        upreg_sources[upreg_sources<span class="sc">$</span>Source <span class="sc">==</span> <span class="st">&quot;WP&quot;</span>, <span class="st">&quot;Number of Terms&quot;</span>]</span>
<span id="cb33-18"><a href="#cb33-18" tabindex="-1"></a>    ),</span>
<span id="cb33-19"><a href="#cb33-19" tabindex="-1"></a>    <span class="st">`</span><span class="at">NoT (Upregulated Only)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(</span>
<span id="cb33-20"><a href="#cb33-20" tabindex="-1"></a>        upreg_sources_only[upreg_sources_only<span class="sc">$</span>Source <span class="sc">==</span> <span class="st">&quot;GO:BP&quot;</span>, <span class="st">&quot;Number of Terms&quot;</span>],</span>
<span id="cb33-21"><a href="#cb33-21" tabindex="-1"></a>        upreg_sources_only[upreg_sources_only<span class="sc">$</span>Source <span class="sc">==</span> <span class="st">&quot;REAC&quot;</span>, <span class="st">&quot;Number of Terms&quot;</span>],</span>
<span id="cb33-22"><a href="#cb33-22" tabindex="-1"></a>        upreg_sources_only[upreg_sources_only<span class="sc">$</span>Source <span class="sc">==</span> <span class="st">&quot;WP&quot;</span>, <span class="st">&quot;Number of Terms&quot;</span>]</span>
<span id="cb33-23"><a href="#cb33-23" tabindex="-1"></a>    ),</span>
<span id="cb33-24"><a href="#cb33-24" tabindex="-1"></a>    <span class="st">`</span><span class="at">NoT (Downregulated)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(</span>
<span id="cb33-25"><a href="#cb33-25" tabindex="-1"></a>        downreg_sources[downreg_sources<span class="sc">$</span>Source <span class="sc">==</span> <span class="st">&quot;GO:BP&quot;</span>, <span class="st">&quot;Number of Terms&quot;</span>],</span>
<span id="cb33-26"><a href="#cb33-26" tabindex="-1"></a>        downreg_sources[downreg_sources<span class="sc">$</span>Source <span class="sc">==</span> <span class="st">&quot;REAC&quot;</span>, <span class="st">&quot;Number of Terms&quot;</span>],</span>
<span id="cb33-27"><a href="#cb33-27" tabindex="-1"></a>        downreg_sources[downreg_sources<span class="sc">$</span>Source <span class="sc">==</span> <span class="st">&quot;WP&quot;</span>, <span class="st">&quot;Number of Terms&quot;</span>]</span>
<span id="cb33-28"><a href="#cb33-28" tabindex="-1"></a>    ),</span>
<span id="cb33-29"><a href="#cb33-29" tabindex="-1"></a>    <span class="st">`</span><span class="at">NoT (Downregulated Only)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(</span>
<span id="cb33-30"><a href="#cb33-30" tabindex="-1"></a>        downreg_sources_only[downreg_sources_only<span class="sc">$</span>Source <span class="sc">==</span> <span class="st">&quot;GO:BP&quot;</span>, <span class="st">&quot;Number of Terms&quot;</span>],</span>
<span id="cb33-31"><a href="#cb33-31" tabindex="-1"></a>        downreg_sources_only[downreg_sources_only<span class="sc">$</span>Source <span class="sc">==</span> <span class="st">&quot;REAC&quot;</span>, <span class="st">&quot;Number of Terms&quot;</span>],</span>
<span id="cb33-32"><a href="#cb33-32" tabindex="-1"></a>        downreg_sources_only[downreg_sources_only<span class="sc">$</span>Source <span class="sc">==</span> <span class="st">&quot;WP&quot;</span>, <span class="st">&quot;Number of Terms&quot;</span>]</span>
<span id="cb33-33"><a href="#cb33-33" tabindex="-1"></a>    )</span>
<span id="cb33-34"><a href="#cb33-34" tabindex="-1"></a>)</span>
<span id="cb33-35"><a href="#cb33-35" tabindex="-1"></a></span>
<span id="cb33-36"><a href="#cb33-36" tabindex="-1"></a><span class="co"># Display the number of terms from each source using kable</span></span>
<span id="cb33-37"><a href="#cb33-37" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(</span>
<span id="cb33-38"><a href="#cb33-38" tabindex="-1"></a>    sources_comparison,</span>
<span id="cb33-39"><a href="#cb33-39" tabindex="-1"></a>    <span class="at">type =</span> <span class="st">&quot;html&quot;</span>,</span>
<span id="cb33-40"><a href="#cb33-40" tabindex="-1"></a>    <span class="at">row.names =</span> <span class="cn">FALSE</span>,</span>
<span id="cb33-41"><a href="#cb33-41" tabindex="-1"></a>    <span class="at">caption =</span> <span class="st">&quot;Number of Terms (NoT) from Differing Methods&quot;</span></span>
<span id="cb33-42"><a href="#cb33-42" tabindex="-1"></a>)</span></code></pre></div>
<table>
<caption>Number of Terms (NoT) from Differing Methods</caption>
<colgroup>
<col width="7%" />
<col width="19%" />
<col width="24%" />
<col width="21%" />
<col width="26%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Source</th>
<th align="right">NoT..Upregulated.</th>
<th align="right">NoT..Upregulated.Only.</th>
<th align="right">NoT..Downregulated.</th>
<th align="right">NoT..Downregulated.Only.</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><a href="GO:BP" class="uri">GO:BP</a></td>
<td align="right">6410</td>
<td align="right">6410</td>
<td align="right">4513</td>
<td align="right">4513</td>
</tr>
<tr class="even">
<td align="left">REAC</td>
<td align="right">1430</td>
<td align="right">1430</td>
<td align="right">1190</td>
<td align="right">1190</td>
</tr>
<tr class="odd">
<td align="left">WP</td>
<td align="right">614</td>
<td align="right">614</td>
<td align="right">502</td>
<td align="right">502</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="interpretation" class="section level1">
<h1>Interpretation</h1>
<p><strong>1. Do the ORA results support conclusions or mechanism
discussed in the original paper?</strong></p>
<p>The results do not immediately support conclusions/mechanisms
discussed in the paper. The authors found that CBD treatment inhibits
SARS-CoV-2 replication in lung epithelial cells<span
class="citation"><sup>4</sup></span>. The most significant hits for the
upregulated genes mainly have to do with responses to the pathways which
growth factor beta (TGF-β) is involved in. However, the most significant
terms which the downregulated genes were associated with had to do with
the cell cycle and replication, mainly chromosome separation and nuclear
division. This could be interpreted as a potential mechanism for the
inhibition of SARS-CoV-2 replication, as the downregulation of these
genes could lead to a decrease in the replication of cells infected with
SARS-CoV-2.</p>
<p><strong>2. Can you find evidence to support some of the results that
you see? How does this evidence support your results?</strong></p>
<p>Literature does mention that TGF-β is involved in the regulation of
the immune system, with effects such as mediating lymphocyte
proliferation<span class="citation"><sup>5</sup></span>. The
upregulation of genes involved in the TGF-β pathway could be interpreted
as a response to the infection of the cells with SARS-CoV-2, with the
immune system “kicking in” at a larger scale from the CBD treatment.</p>
</div>
<div id="references" class="section level1">
<h1>References</h1>
<div id="refs" class="references csl-bib-body">
<div id="ref-atalayAntioxidativeAntiInflammatoryProperties2019"
class="csl-entry">
1. Atalay S, Jarocka-Karpowicz I, Skrzydlewska E. Antioxidative and
<span>Anti-Inflammatory Properties</span> of <span>Cannabidiol</span>.
Antioxidants. 2019 [accessed 2024 Mar 9];9(1):21. doi:<a
href="https://doi.org/10.3390/antiox9010021">10.3390/antiox9010021</a>
</div>
<div id="ref-cassanoCannabisSativaCannabidiol2020" class="csl-entry">
2. Cassano T, Villani R, Pace L, Carbone A, Bukke VN, Orkisz S, Avolio
C, Serviddio G. From <span>Cannabis</span> sativa to
<span>Cannabidiol</span>: <span>Promising Therapeutic Candidate</span>
for the <span>Treatment</span> of <span>Neurodegenerative
Diseases</span>. Frontiers in Pharmacology. 2020 [accessed 2024 Mar
9];11:124. doi:<a
href="https://doi.org/10.3389/fphar.2020.00124">10.3389/fphar.2020.00124</a>
</div>
<div id="ref-bonaccorsoCannabidiolCBDUse2019" class="csl-entry">
3. Bonaccorso S, Ricciardi A, Zangani C, Chiappini S, Schifano F.
Cannabidiol (<span>CBD</span>) use in psychiatric disorders:
<span>A</span> systematic review. Neurotoxicology. 2019;74:282–298.
doi:<a
href="https://doi.org/10.1016/j.neuro.2019.08.002">10.1016/j.neuro.2019.08.002</a>
</div>
<div id="ref-nguyenCannabidiolInhibitsSARSCoV22022" class="csl-entry">
4. Nguyen LC, Yang D, Nicolaescu V, Best TJ, Gula H, Saxena D, Gabbard
JD, Chen S-N, Ohtsuki T, Friesen JB, et al. Cannabidiol inhibits
<span>SARS-CoV-2</span> replication through induction of the host
<span>ER</span> stress and innate immune responses. Science Advances.
2022 [accessed 2024 Mar 9];8(8):eabi6110. doi:<a
href="https://doi.org/10.1126/sciadv.abi6110">10.1126/sciadv.abi6110</a>
</div>
<div id="ref-liTransformingGrowthFactorbeta2006" class="csl-entry">
5. Li MO, Wan YY, Sanjabi S, Robertson A-KL, Flavell RA. Transforming
growth factor-beta regulation of immune responses. Annual Review of
Immunology. 2006;24:99–146. doi:<a
href="https://doi.org/10.1146/annurev.immunol.24.021605.090737">10.1146/annurev.immunol.24.021605.090737</a>
</div>
<div id="ref-ggplot2" class="csl-entry">
6. Wickham H. ggplot2: Elegant graphics for data analysis.
Springer-Verlag New York; 2016. <a
href="https://ggplot2.tidyverse.org">https://ggplot2.tidyverse.org</a>
</div>
<div id="ref-ncbigeo" class="csl-entry">
7. Barrett T, Wilhite SE, Ledoux P, Evangelista C, Kim IF, Tomashevsky
M, Marshall KA, Phillippy KH, Sherman PM, Holko M, et al. <span
class="nocase">NCBI GEO: archive for functional genomics data
sets—update</span>. Nucleic Acids Research. 2012;41(D1):D991–D995. <a
href="https://doi.org/10.1093/nar/gks1193">https://doi.org/10.1093/nar/gks1193</a>.
doi:<a
href="https://doi.org/10.1093/nar/gks1193">10.1093/nar/gks1193</a>
</div>
<div id="ref-biocmanager" class="csl-entry">
8. Morgan M, Ramos M. BiocManager: Access the bioconductor project
package repository. 2023. <a
href="https://CRAN.R-project.org/package=BiocManager">https://CRAN.R-project.org/package=BiocManager</a>
</div>
<div id="ref-knitr" class="csl-entry">
9. Xie Y. Knitr: A comprehensive tool for reproducible research in
<span>R</span>. In: Stodden V, Leisch F, Peng RD, editors. Implementing
reproducible computational research. Chapman; Hall/CRC; 2014.
</div>
<div id="ref-edger" class="csl-entry">
10. Chen Y, Lun AAT, Smyth GK. From reads to genes to pathways:
Differential expression analysis of RNA-seq experiments using rsubread
and the edgeR quasi-likelihood pipeline. F1000Research. 2016;5:1438.
doi:<a
href="https://doi.org/10.12688/f1000research.8987.2">10.12688/f1000research.8987.2</a>
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
