---
title: "Assignment 2"
subtitle: "Analyzing Gene Expression Trends in Cannabidiol (CBD) Treated Cells on SARS-CoV-2 Infection"
author: "Ashenafee Mandefro"
date: "03/09/2024"
bibliography: A2.bib
output:
    html_document:
        toc: true
        toc_float: true
        number_sections: false
        theme: journal
        highlight: tango
        code_folding: hide
        df_print: paged
        self_contained: no
csl: cse.csl
nocite: '@*'
---

# Introduction

## Background {.tabset .tabset-fade}

### Phytocannabinoids

A phytocannabinoid is a cannabinoid that occurs naturally in plants, such as Cannabis *sativa*. The word "phyto" means "plant", and the prefix "phyto-" is used in biochemistry to refer to compounds derived from plants. These compounds are known to occur in several plant species besides cannabis. The distinction with the prefix "phyto-" must be made because the human body also produces cannabinoids, referred to as *endo*cannabinoids due to their endogenous nature.

### Cannabidiol (CBD)

Cannabidiol (CBD) is one of the numerous phytocannabinoids found in the Cannabis (Cannabis sativa) plant. However, it is not responsible for the "high" commonly associated with use of cannabis - that responsibility falls on delta-9 tetrohydrocannabinol (THC). CBD is a non-psychoactive compound, and has been shown to have a wide range of potential therapeutic properties, including anti-inflammatory, antioxidant, and neuroprotective effects [@atalayAntioxidativeAntiInflammatoryProperties2019; @cassanoCannabisSativaCannabidiol2020]. It has showed promise as a possible treatment for a variety of conditions, including epilepsy, anxiety, and chronic pain [@bonaccorsoCannabidiolCBDUse2019].

### SARS-CoV-2

A highly contagious virus, SARS-CoV-2 is the driving force behind the COVID-19 pandemic. It is a novel coronavirus that was first identified in December 2019 in Wuhan, China, and is primarily spread through respiratory droplets. Those who have been infected with the virus can experience a wide range of symptoms, from mild to severe, with the worst outcome being death. Due to its extreme impact on public health, global economy, and daily life, significant amounts of research went into treatment/prevention methods for the virus from 2020 to 2022.

## Primary Research

### Quick Summary of the Paper

[Cannabidiol Inhibits SARS-CoV-2 Replication through Induction of the Host ER Stress and Innate Immune Responses](https://www.science.org/doi/10.1126/sciadv.abi6110) discusses the effects of CBD treatment in SARS-CoV-2 infected cells and mice, through its action of preventing replication of the virus. The researchers found that these effects are apparent in lung epithelial cells which is a critical finding, given that SARS-CoV-2 manifests itself with many respiratory issues[@nguyenCannabidiolInhibitsSARSCoV22022].

### Data Overview

The RNASeq data from this paper is available on the [Gene Expression Omnibus (GEO)](https://www.ncbi.nlm.nih.gov/geo/) under the accession number [GSE168797](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE168797). More information about the data is summarized in the table below:

| Attribute    | Value |
|--------|-------|
| Title  | Cannabidiol Inhibits SARS-CoV-2 Replication through Induction of the Host ER Stress and Innate Immune Responses   |
| Organism | Homo sapiens |
| Paper  | Nguyen, L. C., Yang, D., Nicolaescu, V., Best, T. J., Gula, H., Saxena, D., Gabbard, J. D., Chen, S. N., Ohtsuki, T., Friesen, J. B., Drayman, N., Mohamed, A., Dann, C., Silva, D., Robinson-Mailman, L., Valdespino, A., Stock, L., Suárez, E., Jones, K. A., Azizi, S. A., … Rosner, M. R. (2022). Cannabidiol inhibits SARS-CoV-2 replication through induction of the host ER stress and innate immune responses. Science advances, 8(8), eabi6110. https://doi.org/10.1126/sciadv.abi6110   |
| Samples | 24 |
| Status | Public on Jan 20, 2022   |

# Data Preprocessing

## Data Cleaning

The initial phase of data preprocessing consisted of multiple layers of data cleaning. The raw dataset from GEO had 57 832 entries but after cleaning, that was reduced to 20 438 entries. These >20 000 entries were then used in subsequent steps for data normalization and visualization.

## Data Normalization

The cleaned data was visualized using both a boxplot and a density plot to get a quick overview of the distribution of the data and see if things were indeed normalized successfully.

```{r load_data, eval=TRUE, warning=FALSE, message=FALSE}
# Load the RData file
load("cleaned_dataset.RData")
load("sample_metadata.RData")

lg2_tpm <- log2(cleaned_dataset + 1)
```

```{r boxplot, fig.cap="Boxplot of the normalized data", fig.align="center", out.width="100%"}
# Set the boxplot colors
box_colors <- c(rep("red", 6), rep("blue", 6))

# Plot the normalized data
boxplot(
    lg2_tpm,
    xaxt = "n",
    las = 2,
    main = "Cannabidiol as COVID Treatment RNASeq Boxplot (Normalized)",
    ylab = "Log2 TPM",
    xlab = "Treatment",
    cex = 0.5,
    cex.lab = 0.8,
    col = box_colors
)

# Draw the median line
abline(
    h = median(apply(lg2_tpm, 2, median)),
    col = "green",
    lty = 2,
    lwd = 2
)

# Add x-axis labels manually with a 45 degree rotation
text(
    x = seq_along(lg2_tpm),
    y = par("usr")[3] - 0.2,
    adj = 1,
    srt = 45,
    labels = colnames(lg2_tpm),
    xpd = TRUE,
    cex = 0.8
)

# Add a legend (red - SAR-CoV-2, blue - Mock)
legend(
    "topright",
    legend = c("SARS-CoV-2", "Mock"),
    fill = c("red", "blue")
)
```

```{r densityplot, fig.cap="Density plot of the normalized data", fig.align="center", out.width="100%"}
# Density plot
counts_density <- apply(lg2_tpm, 2, density)
xlim <- 0
ylim <- 0
for (i in seq_along(counts_density)) {
    xlim <- range(c(xlim, counts_density[[i]]$x))
    ylim <- range(c(ylim, counts_density[[i]]$y))
}

cols <- rainbow(length(counts_density))
ltys <- rep(1, length(counts_density))
plot(
    counts_density[[1]],
    xlim = xlim,
    ylim = ylim,
    type = "n",
    ylab = "Smoothing density of log2-CPM",
    main = "",
    cex.lab = 0.85
)
for (i in seq_along(counts_density)) {
    lines(counts_density[[i]], col = cols[i], lty = ltys[i])
}

legend(
    "topright",
    legend = colnames(lg2_tpm),
    col = cols,
    lty = ltys,
    cex = 0.8
)
```

# Differential Expression Analysis

## MDS Plot

```{r mdsplot, fig.cap="MDS plots of the normalized data", fig.align="center", out.width="100%"}
# Create the edgeR container for the count data
d <- edgeR::DGEList(
    counts = as.matrix(cleaned_dataset),
    group = sample_metadata$treatment
)

# Calculate the normalization factors
# d <- edgeR::calcNormFactors(d)

# Set up a two-panel plot layout
par(mfrow = c(1, 2))

# Perform MDS analysis for treatment
mds <- limma::plotMDS(
    d,
    labels = NULL,
    pch = 1,
    col = c("red", "blue")[factor(sample_metadata$treatment)],
)

# Add a legend for treatment
legend(
    "topright",
    legend = levels(factor(sample_metadata$treatment)),
    fill = c("red", "blue")
)

# Perform MDS analysis for infection
mds_infection <- limma::plotMDS(
    d,
    labels = NULL,
    pch = 1,
    col = c("red", "blue")[factor(sample_metadata$infection)],
)

# Add a legend for infection
legend(
    "topright",
    legend = levels(factor(sample_metadata$infection)),
    fill = c("red", "blue")
)
```

## Dispersion and Mean-Variance Plot

```{r dispersion, eval=TRUE, warning=FALSE, message=FALSE, fig.cap="Dispersion Plot", fig.align="center", out.width="100%"}
# Make the layout one panel again
par(mfrow = c(1, 1))

# Create the design matrix
design <- model.matrix(~sample_metadata$treatment)
d <- edgeR::estimateDisp(d, design)

# Plot the dispersion
edgeR::plotBCV(d, col.tagwise = "black", col.common = "red")
```

```{r meanvariance, eval=TRUE, warning=FALSE, message=FALSE, fig.cap="Mean-Variance Plot", fig.align="center", out.width="100%"}
# Plot the mean-variance relationship
edgeR::plotMeanVar(
    d,
    show.raw.vars = TRUE,
    show.tagwise.vars = TRUE,
    NBline = TRUE,
    show.ave.raw.vars = TRUE,
    show.binned.common.disp.vars = TRUE
)
```

```{r volcano, eval=TRUE, warning=FALSE, message=FALSE, fig.cap="Volcano Plot", fig.align="center", out.width="100%"}
# Create the design matrix
design <- model.matrix(~sample_metadata$treatment)

# Fit the model
fit <- edgeR::glmQLFit(d, design)

# Perform the likelihood ratio test
qlf.cbd_vs_dmso <- edgeR::glmQLFTest(
    fit,
    coef = 'sample_metadata$treatmentDMSO'
)

# Plot the volcano plot
results <- data.frame(
    logFC = qlf.cbd_vs_dmso$table$logFC,
    PValue = qlf.cbd_vs_dmso$table$PValue
)

results$significant <- ifelse(results$PValue < 0.05, "yes", "no")

ggplot2::ggplot(
    results,
    ggplot2::aes(x = logFC, y = -log10(PValue), color = significant)
) +
    ggplot2::geom_point(alpha = 0.5) +
    ggplot2::geom_hline(yintercept = -log10(0.05), linetype = "dashed") +
    ggplot2::geom_vline(xintercept = c(-1, 1), linetype = "dashed") +
    ggplot2::labs(
        title = "Volcano Plot",
        x = "Log2 Fold Change",
        y = "-log10 P-Value"
    ) +
    ggplot2::theme_minimal()
```

```{r topgenes, eval=TRUE, warning=FALSE, message=FALSE}

```

## Fitting the Model

```{r fitmodel, eval=TRUE, warning=FALSE, message=FALSE}
# Fit the model
fit <- edgeR::glmQLFit(d, design)

# View(design)

# Normalize the data
norm_counts <- edgeR::cpm(d)

# Perform the likelihood ratio test
qlf.cbd_vs_dmso <- edgeR::glmQLFTest(
    fit,
    coef = ncol(design),
)

knitr::kable(
    edgeR::topTags(qlf.cbd_vs_dmso),
    type = "html",
    row.names = TRUE
)

# Get the results
qlf_output <- edgeR::topTags(
    qlf.cbd_vs_dmso,
    sort.by = "PValue",
    n = nrow(d)
)

length(which(qlf_output$table$PValue < 0.05))
length(which(qlf_output$table$FDR < 0.05))

# Extract the top genes
```

# References
<div id="refs"></div>